One of the first classes in any undergraduate major in biomedical sciences is the histology. The students are tasked with identifying cell types accross a range of tissues, looking for color and shape patterns in hematoxylin-eosin stains. The text-books, such as the one by Junqueira & Carneiro, work as "manuals" (in the Kuhnian sense) perpetuating our paradigms of this set of a couple hundred cell types that define the cells in human beings. 

This anatomy based conceptualization, however, has been constantly challenged by new techniques. The development of techniniques to study cells in the individual level, from multiplexed flow cytometry to patch clamping, opened our eyes for an amazing diversity. With a burst of new categories, novel cell "subtypes" and "families" started to pop up in the literature, enriching our view. This has become specially evident in the past few years, with the explosion of single-cell omics, and the rise of the Human Cell Atlas and the HUBMAP projects. 


Eventhough our technologies diversified, our conceptualization of cell types are still based on century-old histochemical techniques, such as the aforementioned Hematoxylin-eosin staining and Golgi-stainings (imortalized in the drawings of Santiago Ramon y Cajal). We refer to cell types as anatomical entities, as if they were clearly dissectable and fixed in an organism. However, there has not yet been one single definition of cell type that could unite views accross fields, even if for strictly pragmatic purposes.


The advances in computation and the large scale biology, however, requires us to find better alternatives for the answer of what a cell type is. Even if a ontological definition of a "true" cell type is unrealistic, we can strive to find nominal, pragmatic definitions for the pragmatic challenges of large scale biology. Otherwise, how can we proper label single-cell data? How can we judge claims of discovery of cell types? How can we integrate the knowledge from the millions of scientific articles published every year? 


This need for conceptual work is clear, and a number of interesting proposals are arising. Arendt et al has focused on the evolutionary perspective of a cell type: by treating the cell type as an evolutionary unit defined by a Core Regulatory Complex (CoRC) of transcription factors, this view has enabled powerful parallels with the evolution of genes and species. Models of how multicelular life works greatly benefit from concepts such as "sister types", "cell type homology" and "cell type convergence." 

However, iIn an analogous way to how different concepts of species serve better different research programs, our quest to define cell types benefits from an orientation. The challenge of representing cell types in the context of evolution is  conceptually different from representing cell types in the context of biomedical experimentation. The Cell Ontology project and the International Workshop on Cells in ExperimentaL Life Sciences have been doing important groundwork, that base many of the discussions that will be presented in this article. 


In this work, the scope of the quest is the one of research synthesis. How to find a cell type definition that allows rigorous description of biomedical knowledge experiments?

For that goal, this article is divided in the following parts. In Part 1, I will bring a proposal of a set of rules that are sufficient for defining  cell types. In Part 2, I'll propose a small set of names for differentiating main classes of cell types. In Part 3, I will adress the logical consequences of the proposed definitions. And in Part 4, I will discuss the pragmatic challenges for employing such definitions.