One of the first classes in any undergraduate major in life sciences is the histology class. The students are tasked with identifying cell types across various tissues, looking for color and shape patterns in hematoxylin-eosin stains. The text-books, such as the one by Junqueira & Carneiro, work as "manuals" (in the Kuhnian sense), perpetuating our paradigms of what we know about a couple of hundred cell types.

New techniques have challenged this anatomy based conceptualization. From multiplexed flow cytometry to patch clamping, single-cell techniques opened our eyes for an amazing diversity. With a burst of new categories, novel cell "subtypes" and "families" started to pop up in the literature. The rise of new cell types has become especially evident in the past few years, with the explosion of single-cell omics and the creation of the Human Cell Atlas and the HUBMAP projects. 

We advanced our tools, but our concept of "cell type" is still based on century-old histochemical techniques -  such as the Golgi-stains of neurons, immortalized in the drawings of Ramon y Cajal. That implies that the concepts we use are drawn for studies of microanatomy. The connection with anatomy leads to thinking about cell types as anatomical entities as if they were clearly dissectable and fixed in an organism. 

The advances in biology require us to find better answers for how to define a cell type is. The concept might not have a "true" meaning, in a material-realistic sense. Nevertheless, we can strive to find nominal, pragmatic definitions for the pragmatic challenges of large scale biology. Otherwise, how can we properly label single-cell data? How can we judge claims of discovery of cell types? How can we integrate the knowledge from the millions of scientific articles published every year? 

This need for conceptual work is clear, and several proposals for what a cell type are arising. One core line of though is evolutionary: by treating the cell type as an evolutionary unit defined by a Core Regulatory Complex (CoRC) of transcription factors, this view has enabled powerful parallels with the evolution of genes and species. Models of how multicellular life works greatly benefit from concepts such as "sister types", "cell type homology," and "cell type convergence." 

However, in an analogous way to how different concepts of species coexist, our quest to define cell types may take various forms depending on the application. The challenge of representing cell types in the context of evolution is conceptually different from the challenge of representing cell types in biomedical experimentation. In that direction, it is notable the groundwork of the Cell Ontology project and the recurrent ideas proposed at the International Workshop on Cells in ExperimentaL Life Sciences. Their contributions base much of the views here and will be discussed in detail, througout the article. 

The conceptual quest addressed by this work is one of the research synthesis. How to find a cell type definition that allows a rigorous description of biomedical knowledge experiments?

For that goal, the body of the article is divided in 4 parts. In Part 1, I will bring a proposal of a set of rules that are sufficient for defining  cell types. In Part 2, I'll propose a small set of names for differentiating main classes of cell types. In Part 3, I will adress the logical consequences of the proposed definitions. And in Part 4, I will discuss the pragmatic challenges for employing such definitions.