-# A set of 3 + 1 rules for defining a cell type

In an opinion article published in Cell Systems in 2017, researchers presented their views on  the "Conceptual Definition of 'Cell Type' in the Context of a Mature Organism?" [@pmid:28334573]. The opinions vary, and do not converge to a consensus. Many of the scientists see a core role of cells' functions in defining cell types, a slippery road, as the meaning of "function" in biology is elusive [@doi:10.7554/eLife.47014].

In one recent attempt to define cell types for single cell RNA-Seq, Aevermann et al came up with a set of needs: "The minimum set of necessary and sufficient marker genes selectively expressed by the cell type", "A parent cell class in the CL", and "A specimen source description (anatomic structure Ã¾ species)." [@pmid:29590361] They have great merit in defining clear guidelines for marking a cell type. The explicit requirement of markers is reasonable for the field of single-cell RNA-seq, where marker information is abundant. Notably, the Cell Ontology  has even used markers for defining cell types across domains, an approach specially employed for immune cells [@doi:10.1186/1471-2105-12-6] .   

The use of markers, however, leaves us with a conceptual problem: definitions of cell type used by electrophysiologists, or even in manuals of histology classes, are not based on markers. Rigorously, this would leave aside a whole part of what we consider biomedical knowledge. Moreover, gene markers are not defined for cell types that span multiple species. Multispecies markers would require us to explicitly consider homology, adding an extra layer of confusion. 

My pragmatic definition of cell type (for eukaryotic, multicellular organisms) consists of 3 + 1 simple rules. A cell type is any class of cells which _MUST_ be:

- Rigorously defined
- Theoretically useful 
- Identifiable for a defined taxon


And that _SHOULD_ be:

- Logically related to other cell types

The meaning of _MUST_ and _SHOULD_ follow the  the [RFC guidelines for requirement levels ](https://www.ietf.org/rfc/rfc2119.html), where _MUST_ represents an "an absolute requirement of the specification" and _SHOULD_ represents that there may exist valid reasons in particular circumstances to ignore a particular item, but the full implications must be understood and carefully weighed before choosing a different course".  In the following paragraphs, I will clarify the meaning and motivation of the rules.

For rule 1, by "rigorously defined", I mean that a cell can be identified as belonging or not to the class given a set of criteria.  An example of such a rigorous criteria is "expression of the CD3 protein, expression of the CD4 protein, lack of expression of the CD8 protein." This implies the need for using rigorous definitions of what is a "CD3 protein" , and what "expression" means. Similar definitions are used in practice immunology for defining cell types (even though it is not always simple [@doi:10.1186/s12859-019-2725-5] ). In the case of markers, then, any combination of markers (or lack thereof) defines a different _cell type_. In this framework, however, _any_ rigorous definition is aceptable, and small differences are enough for constituting new cell types. For example, a valid definition would be a "big cell" defined as a cell with a length of more than 50 micrometers on any axis. Another valid definition is a "cardiocyte" as any cell in a heart. 

The recognition of multiple valid types of rules is not new. The first Cell Ontology article, in 2005, explicitly acknowledged criteria based on "'function' (for example, electrically_excitable_cell, secretory_cell, photosynthetic_cell), histology (for example, epthelial_cell, mesenchyme_cell), lineage (for example, ectodermal_cell, endodermal_cell) and ploidy (for example, haploid_cell, polyploid_cell)".[@doi:10.1186/gb-2005-6-2-r21]. The Cell Ontology, however, integrates cell types from different phyla. The integrated cell types are useful as points of convergence for databases, or for strucutured human learning. However, for annotation of scientific articles, a definition of the _scope_ of taxons under a cell type may increase the clarity. 

Rule number 2  is, thus,  a spefication of a  rigorousness: one rigorousness criteria is to define the taxons in which a given cell type is expected to have manifestations. We should be able to identify a cell of the type in any individual of the taxon of interest, given the appropriate conditions (e.g. stage of life and biological sex). I will call this set of taxons the _scope_ of the cell type. Note that, as cell types can be defined by function, and functions can converge, a global definition cannot restricted to monophyletic taxa ("clades"). 

Knowing the scope is important to avoid the pitfalls of extrapolation. One recurrent extrapolation is assuming that theories corroborated with mice experiments are valid for other organisms. This extrapolation is an instance of the classic problem of induction, detailed thoroughly in the Logic of Scientific Discovery. A specification of the scope a researcher is referring would make inductional claims explicit and enable proper evaluation of claims.

Currently, the largest authority on cell type definitions, the Cell Ontology, defines _"general cell types"_ which spans multiple animal taxons.[@doi:10.1186/s13326-016-0088-7]  These defitions are conceptually important, but their use for labeling studies plainly is dangerous. It is not safe to assume that a "mouse neutrophil" is simply a "neutrophil" that happens to be in a "mouse".  A definition of scope is essential to tease apart general claims from study-specific claims. 

By providing a specific scope, we can make explicit what are currently implicit predictions. If we take a theory that "the scope of neutrophils is mammals," it explicitly predicts that we expect whales to have neutrophils, for example. If we cannot identify neutrophils after severe tests for any single species, we might agree that the claim has been falsified. Then, we would have to consider this previous concept as unreal, and tailor our understanding of neutrophils to new scopes. 

Rule number 3 is a rule of practical concern. There is a massive amount of "rigorous classes" that one scientist might come up with, due to the combinatorial nature of classes, far outnumbering the reported number of atoms in the observable universe. For that reason, a criterion of usefulness is necessary for deciding when a class of cells is considered a cell type. The simplest criteria of usefulness is one based on the individual: a valid cell type is whatever class any individual rationally finds useful. 

Rule number 4 is one practical extension of the "usefulness" rule: a cell type has to be logically anchored to other cell types for it to be useful. Which means that a definition of a cell class is (for research synthesis conecerns) of lower usefulness if it can't be considered a "subclass" of other cell type. For our practical concerns, all imaginable cell types are subclasses of "cell of eukaryotes". This is presented as a recommendation instead of a reqirement as, in practice, it might be an overhead not strictly necessary for claims of discovery of new cell types and similar tasks.  

The ontological organization is important for integrating knowledge across studies. A "transcriptomically-defined" cell type and a "electrophysiologically-defined" cell type cannot be rigorously said to be the same,  but they can be grouped in a "superclass" that contains cells that match either one or the other criteria. Practically, when describing a cell type, one should make an effort to insert it into the universe of interrelated cell types, even if that implies creating new "superclasses". 

The consequences of this set of criteria will be discussed further in the following sections. 