# A set of 3 + 1 rules for defining a cell type

In an opinion article published in Cell Systems in 2017, researchers presented their views on  the "Conceptual Definition of 'Cell Type' in the Context of a Mature Organism?" [@pmid:28334573]. The opinions vary, and do not converge to a consensus. Many of the scientists see a core role of cells' functions in defining cell types, a slippery road, as the meaning of "function" in biology is elusive [@doi:10.7554/eLife.47014].

In one recent attempt to define cell types for single cell RNA-Seq, Aevermann et al came up with a set of needs: "The minimum set of necessary and sufficient marker genes selectively expressed by the cell type", "A parent cell class in the CL", and "A specimen source description (anatomic structure Ã¾ species)." [@pmid:29590361] They have great merit in defining clear guidelines for marking a cell type. The requirement of markers is reasonable for the field of single-cell RNA-seq, where marker information is abundant. The Cell Ontology  has used markers for defining cell types, an approach specially employed for immune cells [@doi:10.1186/1471-2105-12-6] [@doi:10.1186/s12859-019-2725-5] [@doi:10.1093/bioinformatics/btu807].   

The use of markers, however, leaves us with a conceptual problem: definitions of cell type used by electrophysiologists, or even in the manuals of histology classes, are not based on markers. Rigorously, this would leave aside a whole part of what we consider biomedical knowledge. Moreover, gene markers are not defined for cell types that span multiple species, a problem already discussed on the Cell Ontology report of 2011 [@doi:10.1186/1471-2105-12-6].

My pragmatic definition of cell type (for eukaryotic, multicellular organisms) consists of 3 + 1 simple rules. A cell type is a class of cells that _must_ be:

- Explictly defined
- Theoretically useful 
- Identifiable for a defined taxon


And that _should_ be:

- Logically related to other cell types

For that specification, _must_ represents an "an absolute requirement" and _should_ represents that "there may exist valid reasons in particular circumstances to ignore a particular item" (as per RFC 2119 [@url:https://www.ietf.org/rfc/rfc2119.html])_.

For rule 1, by "explicitly defined", I mean that the cell type needs to be followed by a clear definition, that would allow rational judgements of whether a singular cell belongs or not to the type.  The definition _should_ be _complete_, providing the necessary and sufficient criteria for classification. An example is a cell type defined by "expression of the proteins CD3 and CD4, but lacking CD8." Even though there is still some ambiguity in that definition (see [@doi:10.1186/s12859-019-2725-5] [@doi:10.1093/bioinformatics/btu807] for longer discussions), it already states clear, reasonable criteria. Any combination of markers (or lack thereof) can define a different _cell type_. This extends to _any_  definition, and small differences are enough for constituting new cell types. The degree of rigorousness cannot be decided a priori, as we still do not have a rigorous framework for representing biological knowledge, but we _should_ strive to make definitions as rigorous as possibles. Other examples of what could be explicit definitions: 

- A "big cell", defined a class of cells with a length of more than 50 micrometers on any axis.
- A "human cortical neuron" is any cell in a human cortex that is able of producing an action potential.
- A "leukocyte" is a class of achromatic cells found in animal blood.  

The recognition of multiple valid types of rules is not new. The first Cell Ontology article, in 2005, explicitly acknowledged criteria based on function, histology, lineage and ploidy.[@doi:10.1186/gb-2005-6-2-r21]. These features were combined in the definitions of "species-neutral" cell types[@doi:10.1186/s13326-016-0088-7], arguably useful for integrating databases, or for teaching biology. Gradually, we are acknowledging that we might need more specific classes to characterize experimental biology, leading to definition of species-specific types defined by granular characteristics. [@doi:10.1186/s12859-017-1980-6] [@doi:10.1186/1471-2105-12-6]. 

As an analogy, when describing a new species, besides preserving a type specimen, a taxonomist must cover the species _diagnosis_ - the ways one can tell a species from others. Even though there are standards for format, the taxonomy codes for botanics and zoology do not restrain which characters should be used, as there is a huge diversity of organisms.[@doi:10.1017/CBO9781107415324.004] In the same way, it might be unrealistic to restrict definitions of cell types to a single class of characters, like expression markers. 

Rule number 2  is, thus, an explicit criteria that _must_ be followed when talking about cell types scientifically: we need to define the taxons for which a given cell type is expected to manifest. The cell type, then needs to be findable in any individual of the taxon (or taxons) of interest, given the appropriate conditions (e.g. stage of life and biological sex). The set of taxons covered by a cell type is called here a _taxonomy scope_ (or just _scope_) of the cell type. Note that, as cell types can be defined by function, and functions can converge, the taxonomix scope is not restricted to monophyletic taxa ("clades"). The definition of taxon used here is liberal, as any class of organisms that any researcher identify explicitly as a unit. 

Knowing the scope is important to avoid the pitfalls of extrapolation. One recurrent extrapolation is that theories corroborated by mouse experiments are valid for human cells. This extrapolation is an instance of the classic problem of induction, detailed thoroughly in the Logic of Scientific Discovery. A specification of the scope a researcher is referring to would make inductional claims explicit and enable proper evaluation of claims. It is not safe to assume that a "mouse neutrophil" is simply a "neutrophil" that happens to be in a "mouse".  A definition of scope is essential to tease apart general claims from study-specific claims. 

Rule number 3 is a rule of practical concern. There is a massive amount of "explicit definitions" that one scientist might come up with, due to the combinatorial nature of classes, far outnumbering the reported number of atoms in the observable universe. For that reason, a criterion of usefulness is necessary for deciding when a class of cells is considered a cell type. The simplest criteria of usefulness is one based on the individual: a valid cell type is whatever class any individual rationally finds useful. 

Rule number 4 is one practical extension of the "usefulness" rule: a cell type has to be logically anchored to other cell types for increased usefulness. Which means that a definition of a cell class is (for research synthesis concerns) of lower usefulness if it can't be considered a "subclass" of other cell type. For our practical concerns, all imaginable cell types are subclasses of "cell of eukaryotes". This is presented as a recommendation instead of a requirement as, in practice, it might be an overhead not strictly necessary for claims of discovery of new cell types and similar tasks.  

The ontological organization is important for integrating knowledge across studies. A "transcriptomically-defined" cell type and a "electrophysiologically-defined" cell type are not the same,  but they can be grouped in a "superclass" that contains cells that match either one or the other criteria. Practically, when describing a cell type, one should make an effort to insert it into the universe of interrelated cell types, even if that implies creating new "superclasses". 

The consequences of these set of criteria will be discussed further in the following sections. 