# Practical consequences of the definition


In the previous section, I discussed the logical entailments of accepting the set proposed set of rules as valid. Here, I will extend the pragmatic considerations on using such a system for real-world applications. 

By using the set of rules, we can better evaluate claims of discovering nes cell types. With vast amounts of data and loose definition of cell types, it becomes uncannily easy to claim a new cell type. However, suppose one claims to discover a new "stricto sensu" cell type. In that case, one has to provide enough evidence that cells from this class are identifiable across all individuals of a species. A claim of an "archetype" would require provide evidence in more than one species. Consequently, experiments that only use a specific strain of mice can only claim to discover an infratype.

An example of the discovery of a new "archetype" is the pair of articles published in Nature in 2019 about the newly found "ionocyte", a class of cells in the trachea enriched for expression of genes homologous to the \textit{CFTR} gene. Both studies displayed evidence for such a class in both mouse and human samples, corroborating the existence of an archetype. This discovery has been denominated by both articles as a discovery of a new cell type. 

Another example of cell type discovery is an pioneer article by Villani et al. It describes subclasses of monocytes and dendritic cells in humans, and pragmatically uses markers for their definition. The patients were recruited from "the Boston-based PhenoGenetic project (...) and the Newcastle community." It is arguable that they did not have a random sample of humanity, and the observed results might not hold for different populations. This discovery of infratypes has also been described as a discovery of a new "cell type". 

Thus, by stating the scope of the cell type discovered and rigorously specifying its characteristics, a claim of discovery can be compared in the light of the evidence, very much we have done for centuries for claims of new animal species. 

However, a new problem arises. How to name all these specific cell types? How to we humanely understand with so many "cell types" with such subtle differences? 

For accurately, pragmatically classifying cell types from the perspective of research synthesis, we need rigorous definitions.  These are very specific in nature, and every single empirical article might include several unique technotypes. This makes nomenclature a nightmare. How should we differ B Cells that were selected by slightly different combinations of markers? I avoid this challenge, focusing on the identification of concepts that are computationally useful. 

As described by Sabine Leonelly, the challenges brought up by big data in biology require an advance of our philosophical theories. I argue that the inverse is also true: to advance the theoretical foundations of modern biology, we need to harness the computational tools. The quest for naming cell types becomes simpler when the goal doesn't require human readability at every step. 

Classes in ontologies can have numeric identifiers. By assigning each technotype a Unique Resource Identifier, a URI, and by inserting the URI in a knowledge graph, such as the Cell Ontology or Wikidata, we can start dealing with the complexity of cell types definition across biology. For humans, each article and each dataset could explicitly declare the class it refers by a  name(for example, "neutrophil"), avoiding natural language ambiguities while maintaining readability. 

The Cell Ontology currently holds less than 4.000 cell types. The number of rigorous and useful cell types, however, is considerably larger. By the rules fo deductive logic, whenever you combine two classes ("neutrophil" +"human") you give rise to a third class ("human neutrophil"). We need to consider several classes:  rigorous descriptors, species, anatomical locations, stages of life, biological sexes, strains, stages in the circadian clock and more. The possible number of cell types is of many order of magnitudes higher than currently available. The scale denotes a logistical challenge, as we would require a more significant number of active collaborators in extension and maintenance of such a knowledge base. One way to progress is open systems such as Wikidata, where life scientists can add their 'cell types" with a low entry barrier. The development of such a system is a direction for future research to operationalize the descriptions here. 

The idea of "technotype", if coupled to the possibility for every researcher to craft their "cell type" of interest, solves a major problem of correctly labeling cell types in single-cell experiments. The non-existence of exact matches in CL (even when combined with other ontologies) render it impossible to accurately annotate articles and datasets without incurring in induction issues. As mentioned previously, this would still allow to cross results from different setups, with the gain of explicitly. 

A branch of computational single-cell development has dedicated itself to find tools for labeling single-cell experiments. Some of the approaches have aimed at anchoring the classes used to definitions by the Cell Ontology, both via manual match based on "expert comparison" or via algorithms such as BLAST2CO. By considering technotypes, however, the task conceptually changes not to find a class that "matches" the current experiment, but a candidate class for a "parent" of the described technotype. Albeit subtle, taking this extra layer of consideration can allow more precise conceptualizations of labeling mechanisms and, perhaps, improving the overall performance of such approaches. 