# Practical consequences of the definition


In the previous section, we discussed the logical entailments of accepting the proposed rules as valid. Here, we will extend the pragmatic considerations on using such a system for real-world applications. 

In one recent attempt to define cell types for single cell RNA-Seq, Aevermann et al came up with a set of needs: "The minimum set of necessary and sufficient marker genes selectively expressed by the cell type", "A parent cell class in the CL", and "A specimen source description (anatomic structure Ã¾ species)." [@pmid:29590361] They have great merit in defining clear guidelines for marking a cell type. The requirement of markers is reasonable for the field of single-cell RNA-seq, where marker information is abundant. The Cell Ontology  has used markers for defining cell types, an approach specially employed for immune cells [@doi:10.1186/1471-2105-12-6] [@doi:10.1186/s12859-019-2725-5] [@doi:10.1093/bioinformatics/btu807].   

The use of markers, however, leaves us with a conceptual problem: definitions of cell type used by electrophysiologists, or even in the manuals of histology classes, are not based on markers. Rigorously, this would leave aside a whole part of what we consider biomedical knowledge. Moreover, gene markers are not defined for cell types that span multiple species, a problem already discussed on the Cell Ontology report of 2011 [@doi:10.1186/1471-2105-12-6].

By using our less stringent set of rules, we can better evaluate claims of discover of new cell types. With vast amounts of data and loose definition of cell types, it becomes uncannily easy to claim a new cell type. Converserly, if one explictly claims to discover a new "stricto sensu" cell type, one has to provide enough evidence that cells from this class are identifiable across all individuals of a species. A claim of an "archetype" would require  evidence for existence in more than one species. Consequently, experiments that only use a specific strain of mice have a stronger claim if the expectation is limited to the infratype.

An example of the discovery of a new "archetype" is the pair of articles published in Nature in 2018 [@doi:10.1038/s41586-018-0393-7] [@doi:10.1038/s41586-018-0394-6] about the newly found "ionocyte", a class of cells in the trachea enriched for expression of genes homologous to the _CFTR_ gene. Both studies displayed evidence for such a class in both mouse and human samples, corroborating the existence of an archetype. This discovery of an archetype has been denominated by both articles as a discovery of a new cell type. 

Another example of cell type discovery is a pioneer article by Villani et al [@doi:10.1126/science.aah4573]. It describes subclasses of monocytes and dendritic cells in humans, and pragmatically uses markers for their definition. The patients were recruited from "the Boston-based PhenoGenetic project (...) and the Newcastle community." It is arguable that they did not have a random sample of humanity, and the observed results might not hold for different populations. This discovery of infratypes has also been described as a discovery of a new "cell type". 

An example from the article is the discovery of the  "AS Dendritic cell" (and two subpopulations of it), characterized by expression of the antigens for the proteins AXL and SIGLEC6. This and other cell types are presented in the article  as part of a "Human dendritic cell atlas", generalizing the theory for the whole humanity. However, it is not clear if the population sampled included individuals from different human background. Thus, it is technically possible that the existence of the "cell type" exactly as described might be restricted to some human groups. The jump from technotype (which takes into consideration also descriptors like "healthy" and "age between 25 and 40 years") to infratype ("all humans in this population scope") to cell type _strictu sensu_(all humans) is depicted in the figure @fig:villani and exemplifies the logical flow. 

"Dendritic cells" is one of the cell types most thoroughly modelled by Cell Ontology. [@doi:10.1186/1471-2105-10-70] [@doi:10.1016/j.jbi.2010.01.006]. The current natural language definition of dendritic cell ([CL_0000451](http://purl.obolibrary.org/obo/CL_0000451)) states that a dendritic cell is "A cell of hematopoietic origin, typically resident in particular tissues, specialized in the uptake, processing, and transport of antigens to lymph nodes for the purpose of stimulating an immune response via T cell activation. These cells are lineage negative (CD3-negative, CD19-negative, CD34-negative, and CD56-negative)." The structured definitions are derived from the leukocyte ([CL_0000738](http://purl.obolibrary.org/obo/CL_0000738)) definition, which defines such cells as "achromatic cell of the myeloid or lymphoid lineages capable of ameboid movement." These definitions are not reconcilable to the "dendritic cells" studied by Villani et al's. We have no way of knowing if the cells in their work are "typically resident in particular tissues", "achromatic" or "capable of ameboid movement". That might sound pedantic, and might unfortunately be, but the logical requirements of computational systems leads to both [biocurators](https://www.biocuration.org/community/biocuration-generic-job-description/) and  [computers](https://eloquentjavascript.net/00_intro.html) being seen as pedantic. This high level of precision is necessary to accurately depict not only the complexities of cell types, but the complexities of research settings.

![ Conceptualization of a set of the cell types in Villani et al, 2017 [@doi:10.1126/science.aah4573]. The depicted cell types were manually curated from the article, where they are either implicitly or explicitly mentioned. The set of cell types is not comprehensive, and represent only a small fraction of the concepts handled by the authors. Identifiers for cell types are written in pseudocode based on the Turtle serialization for RDF knowledge graphs (https://www.w3.org/TR/turtle/) and represent valid URIs (described in the database https://celltypes.wiki.opencura.com/wiki/Main_Page). URI: Universal Resource Identifier; RDF: Resource Description Framework.](images/villani_asdcs.png  ){#fig:villani}

Even if we are not yet able to formally represent all the aspects that go into a cell type definition, we can use an explicit "natural language definition" property to define cell types. As David Osumi-Sutherland puts in his 2017 article about cell type classification, there is a "_mismatch between quantified logic, which records assertions about all members of a class, and the messy, noisy reality of biology and the data we collect about it._." [@doi:10.1186/s12859-017-1980-6]. We do not need to have all of biology classified before we deal with cell types. Taking as example in  figure @fig:villani, all cell types treated as "dendritic cells" in the literature are valid subclasses of  the dendritic cell archetype (ct:Q20). Such a subclassing system might lack the power to computationally check the validity of definitions. However, by the principle of minimal commitment [@doi:10.1.1.89.5775], it could already be be a suitable scaffold for representing experimental data (e.g from single-cell transcriptomics) and allow logically robust data integration. 

Sabina Leonelli stated that the challenges brought up by big data in biology require an advance of our philosophical theories [@doi:10.7554/eLife.47381]. We agree, and argue that the inverse is also true: to advance the theoretical foundations of modern biology, we need to harness the power of computational tools. Computational ontologies provide a solution for dealing with complex concepts. Classes in ontologies can have alpha-numeric identifiers. We can, thus assign each technotype a Unique Resource Identifier, a URI, similar to the Cell Ontology (CL) [@doi:10.1186/gb-2005-6-2-r21] [@doi:10.1186/1471-2105-12-6] [@doi:10.1186/s13326-016-0088-7] or Wikidata [@doi:10.7554/eLife.52614] [@https://www.wikidata.org/]. The power of using knowledge bases for integrating knowledge about cell types is becoming clear [@wikidata:Q98665291], and its success relies heavily on precise usage of unique identifiers.

 The quest for names/labels systems for types is related to the quest for defining cell types, and exciting proposals are already arising. [@url:https://arxiv.org/abs/2006.05406] [@doi:10.3389/fnana.2019.00025] Nevertheless, the definition of rules for _naming_ cell types is outside the scope of this article, and abstracted away via semantically void identifiers. A semantically void identifier, in contrast to rule-based numeric identifiers[@doi:10.1002/cyto.a.23346] or natural language names (such as Latin scientific names for species) is resilient to changes in our knowledge. It also floats away from the problems of essence-indicating and role-indicating names  discussed by Rowe and Stone in 1977[@wikidata:Q41052480]. Complementary to being non-descriptive, identifiers can be assigned labels, which can be freely changed, while keeping a persistent Universal Resource Identifier. The  We devoted our effort to refining the logical aspects of cell-type definitions, which might be combined with any commonly agreed naming/labeling system.

The level of "technotype" allows precise labeling of cell types in real world experiments. The non-existence of truly exact matches in CL (even when combined with other ontologies) renders it theoretically impossible to annotate articles and datasets with perfection. The "technotype" gives power to every researcher to craft their "cell type" of interest and connect it to a common network of knowledge.

Specifically, for single-cell transcriptomics, the technotype refines our model for labeling cells (and, consequently, cell clusters). A branch of computational single-cell development has dedicated itself to find tools for labeling single-cell experiments. While some approaches ignore ontologies [@doi:10.1093/nar/gkz543] [@doi:10.1101/2020.01.05.895441], others aim at finding the best class among the Cell Ontology [@https://www.bioconductor.org/packages/release/bioc/vignettes/ontoProc/inst/doc/ontoProc.html#conceptual-overview-of-ontology-with-cell-types] [@doi:10.1101/634097] or MeSH IDs[@doi:10.1101/2020.05.29.124743] [@doi:10.1038/s41467-020-17281-7]. Manual matches to the Cell Ontology have been fed to algorithms such as BLAST2CO [@doi:10.1038/s41467-020-17281-7] and OnClass [@doi:10.1101/810234] to predict best "matches" for a single-cell RNA-seq cells or clusters. However, unless the cells were sampled in the same way across articles, and drawn at random from the same population of individuals, they represent strictly different classes, even if very similar. Thus, we must change the task from finding a "match" to the cells in a given current experiment to finding a  "point of insertion" in an ontological network. By acknowledging these real differences, we can have precise metadata, enabling precise statements and facilitating valid reuse of publically available data. 

An Occan`s razor argument for the use of the technotype is that it makes data integration a simple task. Let a _stricto sensu_ human type X be _completely_ defined by "any cell that expresses the _CD4_ gene, but not the _CD8A_ gene". A definition of this kind allows detection  via both mRNA or proteins, as both are evidence that a cell is _expressing_ a gene. Then, only the information about these two genes is used, and the analyst has only to decide what is the threshold for considering a gene as expressed. All cells that matches such pattern would be assigned to a technotype Y, where Y is a subclass of the _stricto sensu_ human type X. Each cell could receive more than one label (depending on the masks applied to the dataset). A similar rule-based system has been used before to identify different astrocyte states (sic) in post-mortem brains.  [@wikidata:Q89763874]  


Some cells of interest could be missed, especially taking into account dropouts in single-cell. However, with a carefully crafted set of necessary markers, such features can be explicitly accounted for. A cell type definion can employ any logical relation, including "OR". A cell type W can be defined as  any cell that expresses the _MS4A1_ gene OR the _CD79A_ gene". Once such a marker-based definition is decided by the researcher given their research questions, it can be easily tested for any cell, regardless of the size of the dataset. In computational terms, regarding an increasing number of features in the dataset, the mask algorithm scales in big O(1), and regarding the number of cells (provided thad you want to label all cells) in big O(n). If there is no information about the expression of "_MS4A1_ gene OR the _CD79A_ gene", no cell in the dataset can be assigned to _exactly_ this cell type, and another classes need to be employed. Our flexible, yet rigorous, framework for defining cell types can help us to deal with the challenges of varying resolution levels of interest and the scaling large datasets. [doi:10.1186/s13059-020-1926-6]


Such a rule based system enables interpretable data-driven cell-type classifications, in contrast to the previously proposed cell type definitions by reference expression matrices with tens of thousands of genes. Furthermore, it makes the task of _identification_ of cell types almost trivial: you only know what you measure, and that can be checked against an explicit and complete definition (containing the features that are necessary and sufficient for an identification).  Any logical entailments (e.g : a cell that expresses CD3 is achromatic) must be acknowledge explicitly too, if we want to use them for classification and identification. The need of an identification routine for cell-type taxonomies is acknowledge for more than 45 years [@wikidata:Q34065481], and still is a core challenge of human cell-type atlases  [doi:10.1186/s13059-020-1926-6]. The quest for data-driven cell classification is at least as old [@wikidata:Q34065481]. The framework here proposed provides ideas to improve our solutions to both tasks.
