# 4. Practical consequences of the definition of a cell type

In the previous section, we discussed the logical entailments of accepting the proposed rules as valid. 
Here, we extend the pragmatic considerations on using such a system for real-world applications.
In a recent attempt to define cell types for single-cell RNA-Seq, Aevermann et al. came up with a set of needs: "The minimum set of necessary and sufficient marker genes selectively expressed by the cell type," "A parent cell class in the CL (Cell Ontology)," and "A specimen source description (anatomic structure þ species)." [@pmid:29590361]
Their approach has great merit in defining clear guidelines for marking a cell type.
The requirement of markers is reasonable for the field of single-cell RNA-seq, where marker information is abundant.
The Cell Ontology has used markers for defining cell types, an approach employed in particular for immune cells [@doi:10.1186/1471-2105-12-6; @doi:10.1186/s12859-019-2725-5; @doi:10.1093/bioinformatics/btu807].   

However, the use of markers leaves us with a conceptual problem – definitions of "cell type" used by electrophysiologists, or even in the manuals of histology courses, are not based on markers. 
Rigorously adopted, this requirement would leave aside an entire segment of what we consider biomedical knowledge. 
Moreover, gene markers are not defined for cell types that span multiple species, a problem already discussed in the Cell Ontology report of 2011 [@doi:10.1186/1471-2105-12-6]. 
Thus, our set of rules was crafted to accommodate the different ways that people classify cells.

In fact, with so many different takes on the field, vast amounts of data, and loose definitions of cell type, it becomes uncannily easy to claim a new cell type.
Our set of rules may contribute to formalizing cell type discovery.  
If one explicitly claims to have discovered a new _sensu stricto_ cell type, one should provide enough evidence that cells from this class are identifiable across all individuals of a species (given the constraints as age and biological sex).
A claim of an archetype would require evidence of existence in more than one species.
Consequently, experiments that only use a specific strain of mice have a more robust claim if the expectation is limited to the infratype.

An example of the discovery of a new archetype is the pair of articles published in Nature in 2018 [@doi:10.1038/s41586-018-0393-7; @doi:10.1038/s41586-018-0394-6] about the newly found "ionocyte," a class of cells in the trachea enriched for the expression of genes homologous to the human _CFTR_ gene.
Both studies displayed evidence for such a class in both mouse and human samples, corroborating an archetype's existence. This discovery of an archetype has been denominated by both articles as a discovery of a new cell type.

Another example of cell type discovery is found in a pioneering article by Villani et al [@doi:10.1126/science.aah4573].
The authors describe subclasses of monocytes and dendritic cells in humans and pragmatically use markers for their definition.
The patients were recruited from "the Boston-based PhenoGenetic project (…) and the Newcastle community." 
Arguably, they did not have a random sample of humanity, and the observed results might not hold for different populations.
This discovery of infratypes has also been described as the discovery of a new cell type.

An example from the Villani et al. article is the discovery of the "AXL+ SIGLEC6+ AS Dendritic cell".
This and other cell types are presented in the article as part of a "Human dendritic cell atlas", generalizing the theory for the whole of humanity.
The jump from technotype (which takes into consideration also descriptors like "healthy" and "age between 25 and 40 years") to infratype ("all humans in this population scope") to cell type _sensu stricto_ (all humans) is depicted in Figure @fig:villani and exemplifies the logical flow in our proposed framework.

Of note, "dendritic cells" are one of the cell types most thoroughly modeled by the Cell Ontology. [@doi:10.1186/1471-2105-10-70; @doi:10.1016/j.jbi.2010.01.006]
The current definition of the dendritic cell ([CL_0000451](http://purl.obolibrary.org/obo/CL_0000451)) is coupled to the definition of leukocyte ([CL_0000738](http://purl.obolibrary.org/obo/CL_0000738)), which defines it as "An achromatic cell of the myeloid or lymphoid lineages capable of ameboid movement." This definition is not reconcilable with the "dendritic cells" studied by Villani et al. We have no way of knowing if the cells in their work are achromatic or capable of ameboid movement.
That might sound pedantic and might, unfortunately, be so, but the logical requirements of computational systems lead to both biocurators and computers being seen as pedantic. This high level of precision is necessary to accurately depict the complexities of cell types and research settings.

![ Conceptualization of a set of the cell types in Villani et al., 2017 [@doi:10.1126/science.aah4573]. The depicted cell types were manually curated from the article, where they are either implicitly or explicitly mentioned. Identifiers for cell types are written in pseudocode based on the Turtle serialization for RDF (<https://www.w3.org/TR/turtle/>) and represent valid URIs (described in the database <https://celltypes.wiki.opencura.com/wiki/Main_Page>). URI: Universal Resource Identifier; RDF: Resource Description Framework; ct: http://celltypes.wiki.opencura.com/entity/ .](images/villani_asdcs.png  ){#fig:villani}


Even if we cannot represent all the aspects that go into a cell type definition using ontologies, we can use an explicit "natural language definition" property to define cell types.
As David Osumi-Sutherland puts in his 2017 article about cell type classification: there is a "_mismatch between quantified logic, which records assertions about all members of a class, and the messy, noisy reality of biology and the data we collect about it._" [@doi:10.1186/s12859-017-1980-6]. 
Luckily, we do not need to have all the biology formalized before we deal with cell types. 
Taking the example in Figure @fig:villani, all cell types treated as "dendritic cells" in the literature are valid subclasses of the dendritic cell archetype ([ct:Q20](http://celltypes.wiki.opencura.com/entity/)). 
To reach a middleground between natural languages and complete axiomatization, we can use less imprecise Controlled Natural Languages [@wikidata:Q57402167] for better clarity.
Such a system might still lack the power for full computational reasoning, but it could already provide a coherent scaffold for representing experimental data (e.g., from single-cell transcriptomics) and allow logically robust data integration. 
Such a restricted system is more feasible than an "ontology of everything" and would suit the creation of ontologies in accordance with the principle of minimal ontological commitment [@wikidata:Q47303277]). 

The commitment to logical coherence will require us to deal with many more types than we are used to. Given the variety of species on Earth, the complexity of multicellular life, and the diversity of research settings, a count of cell types may far exceed the mark of one million. 
Sabina Leonelli stated that the challenges raised by big data in biology require advancing our philosophical theories [@doi:10.7554/eLife.47381].
We agree and argue that the converse is also true: advancing the theoretical foundations of modern biology requires us to harness the power of computational tools.
Computational ontologies provide a solution for dealing with complex concepts. 
Classes in ontologies can have alpha-numeric identifiers.
We can, thus, assign each technotype a Unique Resource Identifier, a URI, similar to the Cell Ontology (CL)[@wikidata:Q21184168; @doi:10.1186/1471-2105-12-6; @doi:10.1186/s13326-016-0088-7] or in the knowledge graph of  Wikidata [@doi:10.7554/eLife.52614]. 
The power of using knowledge graphs for integrating knowledge about cell types is gaining momentum [@wikidata:Q98665291], and they rely heavily on the precise usage of unique identifiers.

The classification of cells into types and the naming of cell types are parallel tasks. While there has been progress on rules for naming cell types (particularly in neuroscience [@wikidata:Q104247451; @wikidata:Q64065346; @wikidata:Q37656899]), the nomenclature is outside the scope of this article.
Using identifiers/URIs without semantic sense already suffices for our purposes. Semantically void identifiers also help us steer away from the Aristotelian essentialist view upon cell types, as Rowe and Stone discussed in 1977 [@wikidata:Q41052480].
Identifiers can have labels that can be freely changed while keeping a persistent URI. Our effort to refine the logical aspects of cell-type definitions can be combined with any commonly agreed naming/labeling system. 

The URIs at the level of technotype allow precise labeling of cell types in real-world experiments. The technotype annotation empowers researchers to craft their cell type of interest and connect this cell type to a shared knowledge network.
Several single-cell transcriptomics tools try to assign labels to cells.
While some approaches avoid ontologies [@doi:10.1093/nar/gkz543; @wikidata:Q104371272], others  utilize the Cell Ontology [@wikidata:Q101074371; @doi:10.1038/s41467-020-17281-7; @wikidata:Q104057222] or MeSH IDs [@wikidata:Q104371393; @doi:10.1038/s41467-020-17281-7] to identify the most likely cell type label for each cell or cell cluster.
Others go as far as defining markers of interest a priori (like Garnett[@wikidata:Q90003307]]), which goes in the direction of having a explicit definition for the cell types in the study, providing clearer interpretations of results.
There is a clear rush towards tools to "call" cell types. 

However, all of the tools so far only call broad cell types. 
The same name is used for different animals, different life stages or different genders. 
Cell types across this different situations are known to (at least ocasionally) have different properties.
By assingning a general name instead of a more precise one, we are deliberately introducing noise and making it easier to build up nonsense.
By finding very particular technotypes, and then linking them to the general superclasses, the existent algorithms would
For example, instead of claiming that cells from study A and study B are instances of "myeloid dendritic cells", we can be more precise, and say that the cells are instances of either the technotypes A or B and that A and B are subclasses of "myeloid dendritic cell".
The payoff of this extra step is the coherent reuse of publicly available data and a better long term management of conflicting knowledge. 

This flexible yet rigorous framework for defining cell types can help us deal with the challenges of varying resolution levels of interest and scaling large datasets. [doi:10.1186/s13059-020-1926-6] The need for an identification routine for cell-type taxonomies is acknowledge for more than 45 years [@wikidata:Q34065481] and still is a core challenge of human cell-type atlases  [doi:10.1186/s13059-020-1926-6]. The quest for data-driven cell classification is at least as old [@wikidata:Q34065481]. The framework here proposed provides ideas to improve our solutions to both tasks.