# Practical consequences of the definition


In the previous section, I discussed logical entailments of accepting the set proposed set of rules as valid. Here, I will extend the pragmatic considerations on using such a system for real world applications. 

A first pragmatic aplication, already mentioned previously, is that cell type discovery claims can ber organized. With vast amounts of data and loose definition of cell types, it becomes uncannily easy to claim to have discovered a new cell type. If one claims to discover a new "stricto sensu" cell type, one should provide enough evidence that cells from this class are identifiable across all organisms of a species. A claim of an "archetype" should provide evidence in more than one species. And, consequently, experiments that only use a specific strain of mice, can only claim to discover an infratype.

An example of the discovery of a new "archetype" is the pair of articles published in Nature in 2019 about the newly found "ionocyte", a class of cells in the trachea enriched for expression of genes homologous to the \textit{CFTR} gene. Both studies displayed evidence for the existence of such a class in both mouse and human samples, does corroborating the existence of such an archetype. This discovery has been denominated by both articles as a discovery of a new cell type. 

Another example of cell type discovery is an pioneer article by Villani et al that describes subclasses of monocytes and dendritic cells in humans, and pragmatically use markers for their definition. The patients were recruited from "the Boston-based PhenoGenetic project (...) and the Newcastle community." Stricly speaking, it is arguable that they did not have a random sample of humanity, and the observed results might not hold for different populations. Moreover, it is clear that the different "types" described are not being purposed as archetypes. This discovery of infratypes has also been described as a discovery of a new "cell type". 

Thus, by stating the scope of the cell type discovered, and rigorously specifying its characteristics, a claim of discovery can be compared in the light of evidence, very much like it is done for centuries for animal species. However, how can we humanely understand with so many "cell types" with such subtle differences? 

As described by Sabine Leonelly, the challenges brought up by big data in biology require an advance of our philosophycal ground rules. That is a motivation of this work: the new challenges of cell type representation led to the set of rules presented here. I argue that the opposite eay is also true: to advance the theoretical fundations of biology, we need to harness the computational advances. This can be exemplified by the quest of naming specific cell types.


For accurately, pragmatically classifying cell types from the perspective of research synthesis, we need rigorous defintions.  These are very specific in nature, and every single empirical article might include a number of unique technotypes. This makes nomenclature a nightmare. How should we differ B Cells that were selected by a slightly different combinations of markers? Well, the task of finding the correct labels is important for human communication, but can be more easily addressed for describing  research.

Classes in ontologies can have numeric identifiers, which are, in practice, what matters from a computational perspective. By assigning each technotype a Unique Resource Identifier, an URI, and by inserting the URI in a knowledge graph, such as the cell ontology, we can start dealing with the complexity of cell types defintion accross biology. Each article and each dataset, then, can explictly declare once which cless they refer by a term (for example, "neutrophil"), so to avoid the ambiguities of natural language, while maintaining readability. 

The Cell Ontology currently holds less than 4.000 cell types. The number of rigorous and useful cell types, however, is considerably larger. By the rules fo deductive logic, whenever you combine two classes ("neutrophil" +"human", for example) you give rise to a third class ("human neutrophil", for example). If we consider the possible rigorous descriptors, species, anatomical locations, stages of life, biological sexes, strains, stages in the circadian clock and so on, the possible number of cell types is of many order of magnitudes higher. This presents a logistical challenge, which would require a greater number of active collaborators in extension and maintenanceo of such knowledge base. One solution would be open systems such as Wikidata, where life  scientists could add their technotypes with a low entry barrier. The development of such system is a direction for future research  to operationalize the desciprions here. 


The idea of "technotype", if coupled to the possibility for every researcher to craft their "cell type" of interest, solves a major problem of correctly labeling cell types in single cell experiments. The non-existence of exact matches in CL (even when combined with other ontologies) render it impossible to accurately annotate articles and datasets without incurring in induction issues. As mentioned previously, this would still allow to cross results from different setups, with the gain of explicitly. 


A branch of computational single-cell development has dedicated itself to find tools for labeling single cell experiments. Some of the approaches have aimed at anchoring the classes used to defintions by the Cell Ontology, both via manual match based on "expert comparison" or via algorithms such as BLAST2CO. By considering technotypes, however, the task conceptually changes not to find a class that "matches" the current experiment, but a candidate class for a "parent" of the described technotype. Albeit subtle, taking this extra layer of consideration can allow more precise conceptualizations of labeling mechanisms and, perhaps, improving the overall performance of such approaches. 