# Practical consequences of the definition


In the previous section, I discussed the logical entailments of accepting the proposed rules as valid. Here, I will extend the pragmatic considerations on using such a system for real-world applications. 

By using the set of rules, we can better evaluate claims of discover of new cell types. With vast amounts of data and loose definition of cell types, it becomes uncannily easy to claim a new cell type. However, suppose one claims to discover a new "stricto sensu" cell type. In that case, one has to provide enough evidence that cells from this class are identifiable across all individuals of a species. A claim of an "archetype" would require  evidence for existence in more than one species. Consequently, experiments that only use a specific strain of mice can only claim to discover an infratype.

An example of the discovery of a new "archetype" is the pair of articles published in Nature in 2018 [@doi:10.1038/s41586-018-0393-7] [@doi:10.1038/s41586-018-0394-6] about the newly found "ionocyte", a class of cells in the trachea enriched for expression of genes homologous to the _CFTR_ gene. Both studies displayed evidence for such a class in both mouse and human samples, corroborating the existence of an archetype. This discovery has been denominated by both articles as a discovery of a new cell type. 

Another example of cell type discovery is a pioneer article by Villani et al [@doi:10.1126/science.aah4573]. It describes subclasses of monocytes and dendritic cells in humans, and pragmatically uses markers for their definition. The patients were recruited from "the Boston-based PhenoGenetic project (...) and the Newcastle community." It is arguable that they did not have a random sample of humanity, and the observed results might not hold for different populations. This discovery of infratypes has also been described as a discovery of a new "cell type". 

An example from the article is the discovery of the  "AS Dendritic cell" (and two subpopulations of it), characterized by expression of the antigens for the proteins AXL and SIGLEC6. This and other cell types are presented in the article  as part of a "Human dendritic cell atlas", generalizing the theory for the whole humanity. However, it is not clear if the population sampled included individuals from different human background. Thus, it is technically possible that the existence of the "cell type" exactly as described might be restricted to some human groups. The jump from technotype (which takes into consideration also descriptors like "healthy" and "age between 25 and 40 years") to infratype ("all humans in this population scope") to cell type _strictu sensu_(all humans) is depicted in the figure @fig:villani and exemplifies the logical flow. 

![ Conceptualization of a set of the cell types in Villani et al, 2017 [@doi:10.1126/science.aah4573]. The depicted cell types were manually curated from the article, where they are either implicitly or explicitly mentioned. The set of cell types is not comprehensive, and represent only a small fraction of the concepts handled in the article.Identifiers for cell types are written in pseudocode based on the Turtle serialization for RDF knowledge graphs (https://www.w3.org/TR/turtle/) and represent valid URIs (described in the database https://celltypes.wiki.opencura.com/wiki/Main_Page). URI: Universal Resource Identifier; RDF: Resource Description Framework.](images/villani_asdcs.png  ){#fig:villani}

Given the diversity of classes of cell types, stating the _taxonomic scope_ when claiming a discovery (and rigorously definiting its characteristics), a claim of discovery can be compared in the light of the evidence, very much like what we have done for centuries for claims of new animal species. 

However, a new problem arises. How to name all these specific cell types? How to humanely understand so many "cell types" with such subtle differences? 

For accurately, pragmatically classifying cell types from the perspective of research synthesis, we need rigorous definitions.  These are very specific in nature, and every single empirical article might include several unique technotypes. This makes nomenclature a nightmare. Which proverbial names should we use to differ between B Cells that were selected by slightly different combinations of markers?  I avoid this challenge, focusing on the identification of concepts that are computationally useful. 

As described by Sabina Leonelli, the challenges brought up by big data in biology require an advance of our philosophical theories [@doi:10.7554/eLife.47381]. I argue that the inverse is also true: to advance the theoretical foundations of modern biology, we need to harness the computational tools. The quest for naming cell types becomes simpler when the goal doesn't require human readability at every step. Instead, by harnessing the computer power to record complex definitions, we can focus on higher level abstractions (not unlike using contact names to access hundreds of phone numbers).  

Classes in ontologies can have numeric identifiers. By assigning each technotype a Unique Resource Identifier, a URI, and by inserting the URI in a knowledge graph, such as the Cell Ontology [@doi:10.1186/gb-2005-6-2-r21] [@doi:10.1186/1471-2105-12-6] [@doi:10.1186/s13326-016-0088-7] or Wikidata [@doi:10.7554/eLife.52614] [@https://www.wikidata.org/], we can start dealing with the complexity of cell types definition across biology. For humans, each article and each dataset could explicitly declare the class it refers by a  name (for example, "neutrophil"), avoiding natural language ambiguities while maintaining readability. 

The Cell Ontology currently holds less than 4.000 cell types. The number of rigorous and useful cell types, however, is considerably larger. By the rules fo deductive logic, whenever you combine two classes ("neutrophil" +"human") you give rise to a third class ("human neutrophil"). We need to consider several classes:  rigorous descriptors, species, anatomical locations, stages of life, biological sexes, strains, stages in the circadian clock and more. The possible number of cell types is of many order of magnitudes higher than currently available. The scale denotes a logistical challenge, as we would require a more significant number of active collaborators in extension and maintenance of such a knowledge base. One way to progress is open systems such as Wikidata, where life scientists can add their 'cell types" with a low entry barrier. The development of such a system is a direction for future research to operationalize the descriptions here. 

The idea of "technotype", if coupled to the possibility for every researcher to craft their "cell type" of interest, can solve the problem of correctly labeling cell types in single-cell experiments. The non-existence of exact matches in CL (even when combined with other ontologies) renders it theoretically impossible to annotate articles and datasets without incurring in induction issues. As mentioned in the previous section, by having a knowledge connecting the concepts, we would still be able to compare results from different researchers, but now explicitly stating the level of abstraction in which they can be compared.

A branch of computational single-cell development has dedicated itself to find tools for labeling single-cell experiments. While some approaches avoid using ontologies [@doi:10.1093/nar/gkz543] [@doi:10.1101/2020.01.05.895441], others aim at anchoring the classes to the Cell Ontology [@https://www.bioconductor.org/packages/release/bioc/vignettes/ontoProc/inst/doc/ontoProc.html#conceptual-overview-of-ontology-with-cell-types] or to MeSH IDs[@doi:10.1101/2020.05.29.124743] [@doi:10.1038/s41467-020-17281-7] via manual match based on "expert comparison" of labels. These manual matches have been fed to algorithms such as BLAST2CO [@doi:10.1038/s41467-020-17281-7] to predict best "matches" for a single-cell cluster. Even though cell-type prediction is an exciting area of research, its current approach is inherently imprecise. Unless the cells were sampled in the same way across articles, drawn at random from the same population of individuals, they represent strictly different classes, even if very similar.

 As mentioned by Sartivijai, Diehl and Yongqun, "Ontology classes should always represent certainty, and not probability."[@doi:10.1186/s12859-019-2721-9]. Considering the "technotype" level for single-study cell types improves that precision. We can change the task from finding a "match" to the cells in a given current experiment to finding a candidate class for a "point of insertion" of the described technotype, even directly (in a parent-child relation to previous class), or by creating a new super class (sibling relation to previous class). Albeit subtle, this extra layer of consideration allows more precise conceptualizations of labeling mechanisms, a theoretical advance  that can  pave the way for improvements in automatic classification algorithms.