<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Tiago Lubiana" />
  <meta name="author" content="Helder Takashi Imoto Nakaya (pending approval)" />
  <meta name="dcterms.date" content="2020-11-20" />
  <meta name="keywords" content="cell type, nomenclature, classification" />
  <title>Towards a pragmatic definition of cell type</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/master/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="Towards a pragmatic definition of cell type" />
  <meta name="citation_title" content="Towards a pragmatic definition of cell type" />
  <meta property="og:title" content="Towards a pragmatic definition of cell type" />
  <meta property="twitter:title" content="Towards a pragmatic definition of cell type" />
  <meta name="dc.date" content="2020-11-20" />
  <meta name="citation_publication_date" content="2020-11-20" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Tiago Lubiana" />
  <meta name="citation_author_institution" content="Department of Clinical and Toxicological Analyses, School of Pharmaceutical Sciences, University of São Paulo, São Paulo, Brazil" />
  <meta name="citation_author_orcid" content="0000-0003-2473-2313" />
  <meta name="citation_author" content="Helder Takashi Imoto Nakaya (pending approval)" />
  <meta name="citation_author_institution" content="Department of Clinical and Toxicological Analyses, School of Pharmaceutical Sciences, University of São Paulo, São Paulo, Brazil" />
  <meta name="citation_author_orcid" content="0000-0001-5297-9108" />
  <link rel="canonical" href="https://lubianat.github.io/technotype/" />
  <meta property="og:url" content="https://lubianat.github.io/technotype/" />
  <meta property="twitter:url" content="https://lubianat.github.io/technotype/" />
  <meta name="citation_fulltext_html_url" content="https://lubianat.github.io/technotype/" />
  <meta name="citation_pdf_url" content="https://lubianat.github.io/technotype/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://lubianat.github.io/technotype/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://lubianat.github.io/technotype/v/af47013b89f008fc8e1df794d570b51f6a40adcd/" />
  <meta name="manubot_html_url_versioned" content="https://lubianat.github.io/technotype/v/af47013b89f008fc8e1df794d570b51f6a40adcd/" />
  <meta name="manubot_pdf_url_versioned" content="https://lubianat.github.io/technotype/v/af47013b89f008fc8e1df794d570b51f6a40adcd/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Towards a pragmatic definition of cell type</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://lubianat.github.io/technotype/v/af47013b89f008fc8e1df794d570b51f6a40adcd/">permalink</a>)
was automatically generated
from <a href="https://github.com/lubianat/technotype/tree/af47013b89f008fc8e1df794d570b51f6a40adcd">lubianat/technotype@af47013</a>
on November 20, 2020.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Tiago Lubiana</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-2473-2313">0000-0003-2473-2313</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/lubianat">lubianat</a><br>
<small>
Department of Clinical and Toxicological Analyses, School of Pharmaceutical Sciences, University of São Paulo, São Paulo, Brazil
</small></p></li>
<li><p><strong>Helder Takashi Imoto Nakaya (pending approval)</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-5297-9108">0000-0001-5297-9108</a><br>
<small>
Department of Clinical and Toxicological Analyses, School of Pharmaceutical Sciences, University of São Paulo, São Paulo, Brazil
</small></p></li>
</ul>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<h1 id="introduction">Introduction</h1>
<p>One of the basic classes in any undergraduate major in life sciences is the histology class. The students are tasked with identifying cel types across various tissues, looking for color and shape patterns in hematoxylin-eosin stains. The text-books, such as the one by Junqueira &amp; Carneiro<span class="citation" data-cites="isbn:9788527723114">[<span class="citeproc-not-found" data-reference-id="isbn:9788527723114"><strong>???</strong></span>]</span>, work as manuals (in the Kuhnian sense)<span class="citation" data-cites="E78joy5H">[<a href="#ref-E78joy5H" role="doc-biblioref">1</a>]</span> that perpetuate the paradigms of what we know about a couple hundred cell types.</p>
<p>The concept of “cell type”, thus, is still based on century-old histochemical techniques - such as the Golgi-stains of neurons immortalized by Ramon y Cajal<span class="citation" data-cites="H4OiWiM2">[<a href="#ref-H4OiWiM2" role="doc-biblioref">2</a>]</span>. The histological influence is noticeable even in names given to cell types: “erithrocytes”, “eosinophil”, “basophils”, “oxyphilic cell of the thyroid”. The concepts we use are drawn from studies of microanatomy. The connection with anatomy leads to thinking about cell types as anatomical entities as if they were clearly dissectable and fixed in an organism. This may be the reason why attempts to quantify cell types use the scale of “hundreds” of human cell types <span class="citation" data-cites="jePJgR0H">[<a href="#ref-jePJgR0H" role="doc-biblioref">3</a>]</span> <span class="citation" data-cites="V85zvNGa">[<a href="#ref-V85zvNGa" role="doc-biblioref">4</a>]</span> <span class="citation" data-cites="H4wZSl5d">[<a href="#ref-H4wZSl5d" role="doc-biblioref">5</a>]</span>.</p>
<p>New techniques have challenged this anatomy based conceptualization. From flow cytometry to patch clamping, to single-cell RNA-seq, we saw a burst of new categories, and novel cell “subtypes” and “families” popped up in the literature. The bursting intensified in the past few years, with the rise of projects to characterize <em>all</em> human cell types, like the Human Cell Atlas and HUBMAP <span class="citation" data-cites="vk9ZInF3">[<a href="#ref-vk9ZInF3" role="doc-biblioref">6</a>]</span> <span class="citation" data-cites="167nEKPOQ">[<a href="#ref-167nEKPOQ" role="doc-biblioref">7</a>]</span>.</p>
<p>The advances in biology require us to find better answers for how to define a cell type. The concept might not have a “true” meaning, in a philosophical-realistic sense. Nevertheless, we can strive to find nominal, pragmatic definitions for the pragmatic challenges of large scale biology. Otherwise, how can we precisely label single-cell data? How can we formalize the discovery of new cell types? How can we integrate the knowledge from the millions of scientific articles published every year?</p>
<p>The need for conceptual advance is perceived by the community <span class="citation" data-cites="5QxCvchL">[<a href="#ref-5QxCvchL" role="doc-biblioref">8</a>]</span> <span class="citation" data-cites="sWv3okaH">[<a href="#ref-sWv3okaH" role="doc-biblioref">9</a>]</span> <span class="citation" data-cites="19yYJWEkz">[<a href="#ref-19yYJWEkz" role="doc-biblioref">10</a>]</span> <span class="citation" data-cites="1DYAsd4eB">[<a href="#ref-1DYAsd4eB" role="doc-biblioref">11</a>]</span> <span class="citation" data-cites="7zWO5k6T">[<a href="#ref-7zWO5k6T" role="doc-biblioref">12</a>]</span> <span class="citation" data-cites="TupzTvtA">[<a href="#ref-TupzTvtA" role="doc-biblioref">13</a>]</span> <span class="citation" data-cites="w0Woz7JE">[<a href="#ref-w0Woz7JE" role="doc-biblioref">14</a>]</span> , and new perspectives are arising. One core line of thought is evolutionary: the cell type as an evolutionary unit defined by a Core Regulatory Complex (CoRC) of transcription factors. That definition enables the drawing of parallels between the evolution of other biological entities (such as genes, proteins and species) to the evolution of cell types. Models of how multicellular life works greatly benefit from concepts such as “sister types” (cell types that diverged from a single ancestor), “cell type homology,” (cell types in different species that share a common evolutionary origin) and “cell type convergence” (cell types that execute similar functions, but which are not directly evolutionarily related). <span class="citation" data-cites="O95WN4x0">[<a href="#ref-O95WN4x0" role="doc-biblioref">15</a>]</span> <span class="citation" data-cites="ogAGO9KH">[<a href="#ref-ogAGO9KH" role="doc-biblioref">16</a>]</span></p>
<p>However, as much as different concepts of species coexist <span class="citation" data-cites="g74dZjOM">[<a href="#ref-g74dZjOM" role="doc-biblioref">17</a>]</span>, our quest to define cell types may take various forms. The challenge of representing cell types in the context of evolution is conceptually different from the challenge of representing cell types in biomedical experimentation. In that second direction, the groundwork of the Cell Ontology <span class="citation" data-cites="1CCBSogYG">[<a href="#ref-1CCBSogYG" role="doc-biblioref">18</a>]</span> <span class="citation" data-cites="gAysgnsS">[<a href="#ref-gAysgnsS" role="doc-biblioref">19</a>]</span> <span class="citation" data-cites="17U3pGEjv">[<a href="#ref-17U3pGEjv" role="doc-biblioref">20</a>]</span> project and the International Workshop on Cells in Experimental Life Sciences<span class="citation" data-cites="doi:10.1186/s12859-017-1976-2">[<span class="citeproc-not-found" data-reference-id="doi:10.1186/s12859-017-1976-2"><strong>???</strong></span>]</span> <span class="citation" data-cites="rK8KD0us">[<a href="#ref-rK8KD0us" role="doc-biblioref">21</a>]</span> are notable. Their contributions base much of the views here and will be discussed in detail throughout the article.</p>
<p>The conceptual quest addressed by this work is one of the research synthesis and is summarized in the following question: which definition of cell type optimizes the rigorous description biomedical experiments?</p>
<p>We chose to use the term “cell type” to emphasize the focus on types as classes (or “kinds”) in contrast to real-world objects. The notion of “cell state” is used both to describe classes (e.g activated T-cell) as well as for real world observations(e.g. the current state of a specific cell). One can argue that 2 cells can never be in the same state, in an extension of to the Heraclitian idea of not being able to step twice in the same river <span class="citation" data-cites="EDf5c4SG">[<a href="#ref-EDf5c4SG" role="doc-biblioref">22</a>]</span>. Other similar notions, such as a “cell set”, “cell population” and “cell cluster” can also reminisce of a specific, countable group of cells, frequently from a same experiment. The term “cell type” is used for relatively invariant classes of cells, with common characteristics.</p>
<p>The term “cell class” is also used in the literature, and would be a suitable synonym for our notion of cell type, as the main goal here is to refine the human-based theoretical classes. Classes that we can <em>instantiate</em>, i.e. assign to an observation of any real cell, in the same way we assign the class <em>Homo sapiens</em> to each and every human. The term “cell identity has also been suggested for avoiding the cell type/cell state dilemma <span class="citation" data-cites="cJ3Q82kT">[<a href="#ref-cJ3Q82kT" role="doc-biblioref">23</a>]</span>. The notion of identity is however slightly different of the idea of class. We nevertheless opted to frame our work around”cell type" due to its historical usage and familiarity for the life sciences community.</p>
<p>For that goal, the body of the article is divided in 4 parts. In Part 1, we will bring a proposal of a set of rules that are sufficient for defining cell types. In Part 2, we will propose a small set of names for differentiating main classes of cell types. In Part 3, we will address the logical consequences of the proposed definitions. And in Part 4, we will discuss the pragmatic challenges for employing such definitions.</p>
<h1 id="a-set-of-3-1-rules-for-defining-a-cell-type">A set of 3 + 1 rules for defining a cell type</h1>
<p>In an opinion article published in Cell Systems in 2017, researchers presented their views on the “Conceptual Definition of ‘Cell Type’ in the Context of a Mature Organism?” <span class="citation" data-cites="5QxCvchL">[<a href="#ref-5QxCvchL" role="doc-biblioref">8</a>]</span>. The opinions vary, and do not converge to a consensus. Many of the scientists see a core role of cells’ functions in defining cell types, a slippery road, as the meaning of “function” in biology is elusive <span class="citation" data-cites="131Mt6Qd2">[<a href="#ref-131Mt6Qd2" role="doc-biblioref">24</a>]</span>.</p>
<p>Our pragmatic definition of cell type (for eukaryotic, multicellular organisms) consists of 3 + 1 simple rules. A cell type is a class of cells that <em>must</em> be:</p>
<ul>
<li>Explictly defined</li>
<li>Theoretically useful</li>
<li>Identifiable for a defined taxon</li>
</ul>
<p>And that <em>should</em> be:</p>
<ul>
<li>Hierarchically related to other cell types</li>
</ul>
<p>For that specification, <em>must</em> represents an “an absolute requirement” and <em>should</em> represents that “there may exist valid reasons in particular circumstances to ignore a particular item” (as per RFC 2119 <span class="citation" data-cites="56biRUeN">[<a href="#ref-56biRUeN" role="doc-biblioref">25</a>]</span>)_.</p>
<p>For rule 1, by “explicitly defined”, we mean that the cell type needs to be followed by a clear definition, that would allow rational judgements of whether a singular cell belongs or not to the type. The definition <em>should</em> be <em>complete</em>, providing the necessary and sufficient criteria for classification. An example is a cell type defined by “expression of the proteins CD3 and CD4, but lacking CD8.” Even though there is still some ambiguity in that definition (see <span class="citation" data-cites="9UTyLFor">[<a href="#ref-9UTyLFor" role="doc-biblioref">26</a>]</span> <span class="citation" data-cites="17uICvtFm">[<a href="#ref-17uICvtFm" role="doc-biblioref">27</a>]</span> for longer discussions), it already states clear, reasonable criteria. Any combination of markers (or lack thereof) can define a different <em>cell type</em>. This extends to <em>any</em> definition, and small differences are enough for constituting new cell types. The degree of rigorousness cannot be decided a priori, as we still do not have a rigorous framework for representing biological knowledge, but we <em>should</em> strive to make definitions as rigorous as possibles. Other examples of what could be explicit definitions:</p>
<ul>
<li>A “big cell”, defined a class of cells with a length of more than 50 micrometers on any axis.</li>
<li>A “human cortical neuron” is any cell in a human cortex that is able of producing an action potential.</li>
<li>A “leukocyte” is a class of achromatic cells found in animal blood.</li>
</ul>
<p>The recognition of multiple valid types of rules is not new. The first Cell Ontology article, in 2005, explicitly acknowledged criteria based on function, histology, lineage and ploidy.<span class="citation" data-cites="1CCBSogYG">[<a href="#ref-1CCBSogYG" role="doc-biblioref">18</a>]</span>. These features were combined in the definitions of “species-neutral” cell types<span class="citation" data-cites="17U3pGEjv">[<a href="#ref-17U3pGEjv" role="doc-biblioref">20</a>]</span>, arguably useful for integrating databases, or for teaching biology. Gradually, we are acknowledging that we might need more specific classes to characterize experimental biology, leading to definition of species-specific types defined by granular characteristics. <span class="citation" data-cites="KJDots8y">[<a href="#ref-KJDots8y" role="doc-biblioref">28</a>]</span> <span class="citation" data-cites="gAysgnsS">[<a href="#ref-gAysgnsS" role="doc-biblioref">19</a>]</span>.</p>
<p>As an analogy, when describing a new species, besides preserving a type specimen, a taxonomist must cover the species <em>diagnosis</em> - the ways one can tell a species from others. Even though there are standards for format, the taxonomy codes for botanics and zoology do not restrain which characters should be used, as there is a huge diversity of organisms.<span class="citation" data-cites="wpSlfce3">[<a href="#ref-wpSlfce3" role="doc-biblioref">29</a>]</span> In the same way, it might be unrealistic to restrict definitions of cell types to a single class of characters, like expression markers.</p>
<p>Rule number 2 is, thus, an explicit criteria that <em>must</em> be followed when talking about cell types scientifically: we need to define the taxons for which a given cell type is expected to manifest. The cell type, then needs to be findable in any individual of the taxon (or taxons) of interest, given the appropriate conditions (e.g. stage of life and biological sex). The set of taxons covered by a cell type is called here a <em>taxonomy scope</em> (or just <em>scope</em>) of the cell type. Note that, as cell types can be defined by function, and functions can converge, the taxonomix scope is not restricted to monophyletic taxa (“clades”). The definition of taxon used here is liberal, as any class of organisms that any researcher identify explicitly as a unit.</p>
<p>Knowing the scope is important to avoid the pitfalls of extrapolation. One recurrent extrapolation is that theories corroborated by mouse experiments are valid for human cells. This extrapolation is an instance of the classic problem of induction, detailed thoroughly in the Logic of Scientific Discovery. A specification of the scope a researcher is referring to would make inductional claims explicit and enable proper evaluation of claims. It is not safe to assume that a “mouse neutrophil” is simply a “neutrophil” that happens to be in a “mouse”. A definition of scope is essential to tease apart general claims from study-specific claims.</p>
<p>Rule number 3 is a rule of practical concern. There is a massive amount of “explicit definitions” that one scientist might come up with, due to the combinatorial nature of classes, far outnumbering the reported number of atoms in the observable universe. For that reason, a criterion of usefulness is necessary for deciding when a class of cells is considered a cell type. The simplest criteria of usefulness is one based on the individual: a valid cell type is whatever class any individual rationally finds useful.</p>
<p>Rule number 4 is one practical extension of the “usefulness” rule: a cell type has to be hierarchically anchored to other cell types for increased usefulness. Which means that a definition of a cell class is (for research synthesis concerns) of lower usefulness if it can’t be considered a “subclass” of other cell type. For our practical concerns, all imaginable cell types are subclasses of “cell of eukaryotes”. This is presented as a recommendation instead of a requirement as, in practice, it might be an overhead not strictly necessary for claims of discovery of new cell types and similar tasks.</p>
<p>The ontological organization is important for integrating knowledge across studies. A “transcriptomically-defined” cell type and a “electrophysiologically-defined” cell type are not the same, but they can be grouped in a “superclass” that contains cells that match either one or the other criteria. Practically, when describing a cell type, one should make an effort to insert it into the universe of interrelated cell types, even if that implies creating new “superclasses”.</p>
<p>The consequences of these set of criteria will be discussed further in the following sections.</p>
<h1 id="naming-classes-of-cell-types">Naming classes of cell types</h1>
<p>In parallel to the criteria raised on part 1,I propose a set of naming conventions for different classes of cell types, to facilitate communication. Much of the literature mixes cell types in one species (e.g., when dealing with a cell type as an evolutionary unit) and multispecies (e.g., in the cell ontology). It is arguably useful to distill these different concepts into their own names.. Given the importance of the concept of species in biological classification <span class="citation" data-cites="xDF1wKqQ">[<a href="#ref-xDF1wKqQ" role="doc-biblioref">30</a>]</span> <span class="citation" data-cites="1ER6m3I7b">[<a href="#ref-1ER6m3I7b" role="doc-biblioref">31</a>]</span>, we derive a “species-centric” view on the naming of classes of cell types. The three classes we propose are:</p>
<ul>
<li>Archetypes, for when the <em>taxonomic scope</em> of the type is beyond the level of species. For example, “mammal neutrophils.”</li>
<li><em>Stricto sensu</em> cell types, for when the <em>taxonomic scope</em> of the type corresponds to a single species. For example, “human neutrophils.”</li>
<li>Infratypes, for when the <em>taxonomic scope</em> is below the level of species. For example, considering the mouse strain “C57BL/6J” <span class="citation" data-cites="eM5m3jNA">[<a href="#ref-eM5m3jNA" role="doc-biblioref">32</a>]</span>, “C57BL/6J neutrophils”.</li>
</ul>
<p>By adopting a more precise vocabulary, we can flesh out misunderstandings and communicate clearly. At the level of individual scientific experiments, we usually work at the “infratype” level: the samples come only from a subpopulation of the species of interest, and cannot be assumed to be “randomly sampled” from all individuals. This has important practical considerations for, once more, avoiding failing implicitly for the problem of induction.</p>
<p>In addition, in individual experiments, we work with cells of very specific classes. They are not only infratypes, but very specific infratypes, defined by non-random research setups and pragmatic choices. For example, we might call “CD4 T cells” what are actually CD3+, CD4+, CD8+ cells from the axillary lymph node of 2-month-old chow-fed female C57BL6/J mice from the mouse-house of the Institute of Biochemistry of the University of São Paulo collected in several mornings around 10 pm. Albeit really specific, all the mentioned facets (markers, anatomical location, age, biological sex, strain, housing conditions, circadian clock and diet) are known to alter what we know about cell types. Thus, we benefit from using a name for these very specific classes:</p>
<ul>
<li>Technotype: A specific, experimentally defined cell type, which harbors on its definitions the precise conditions of the cells sampled.</li>
</ul>
<p>Even if really specific, a technotype is still a class. Unless a study used only one single-cell, it likely contained some sampling method. Samples are sampled from a specific population, for which hypothesis are actually tested. This is the most "granular’ cell type in our pragmatic view for research synthesis. This is the type that can be strictly annotated in single-cell RNA-seq analysis, for example.</p>
<p>Single claims are made and tested for technotypes, and the claims can be logically combined in “upper” ontological levels for making claims with a higher degree of universality. The propagation of knowledge to upper levels cannot be implicit. (see Yarkoni 2020 for an analogous problem in the psychological sciences <span class="citation" data-cites="b2FgShV8">[<a href="#ref-b2FgShV8" role="doc-biblioref">33</a>]</span>). As defended by Popper, knowledge should travel “quasi-inductionally” by fostering hypothesis with higher degrees of generality, which can then be tested for the more universal class <span class="citation" data-cites="sFrgW0cH">[<a href="#ref-sFrgW0cH" role="doc-biblioref">34</a>]</span>.</p>
<h1 id="logical-consequences-of-the-definition">Logical consequences of the definition</h1>
<p>One logical consequence of the rules is that concepts of “subtype” become redundant with “cell type.” A “subtype,” then, is a concept that only makes sense when talking about classes with different degrees of universality. Thus, claims to discover new cell “subtypes” or “types” differ only stylistically and can be considered indistinguishable in the perspective of research synthesis.</p>
<p>The cell type as defined here, includes both persistent classes of cells - the traditional cell types - and transient, fugacious classes of cells - often also called states or even programs. Even though such distinctions are important topic for theoretical research, it might not a requirement for representing biomedical experiments. In fact, the fuzzy border of activity (“traditional cell state”) and identity (“traditional cell types”) gene expression programs is acknowledged by the community <span class="citation" data-cites="3H80HarF">[<a href="#ref-3H80HarF" role="doc-biblioref">35</a>]</span>. Aviv Regev, a co-chair of the Human Cell Atlas mentioned in a recent HCA Latin America Meeting, “in a sense, for any cell, at any moment in time, when you are profiling it, what you are seeing is its state. (…) There is the overall state of the cell, which is made of many different programs, some of which represent this more invariant part of the cell’s identity that we tend to call a type, whereas others are more variable and they can change for a cell of the same type” (https://youtu.be/1hOwgTH6vZY?list=PLkef4SGmngdbGgJIQh7xDI-lgMZMsl-YJ&amp;t=1089). The rules proposed here aim at harmonizing these ideas, by explicitly allowing cell type definitions to include descriptors related to activity, identity, or a mix of both.</p>
<p>One example of this entailment is that the class “human cells in metaphase of mitosis” can be considered a cell type, as they can be explictly defined and restricted to a taxon. Even though “metaphase” itself is a biological process, we can describe all cells executing this process as a single cell type.</p>
<p>However, does a dividing fibroblast stop being a fibroblast, even if temporarily? Again, we do not aim to answer this in a philosophical-ontological sense. Pragmatically, if the explicit definition used for fibroblast (e.g., expression of a marker) still holds during duplication, this cell can be assigned to two classes that are not hierarchically related: “fibroblasts” and “doubling cells”. It is, thus, essential to consider that cells can belong to at least two disjunct classes.</p>
<p>If cells can be assigned to multiple classes that are not hierarchically related, it is not possible to annotate cell types using a single taxonomic tree, in which each concept would be represented by a single node with one (and only one) direct parent node. This is conflicting with attempts to classify cell-types using single hierarchies in the form of a tree <span class="citation" data-cites="I4hT5zr7">[<a href="#ref-I4hT5zr7" role="doc-biblioref">36</a>]</span> <span class="citation" data-cites="HixFZJ8X">[<a href="#ref-HixFZJ8X" role="doc-biblioref">37</a>]</span> <span class="citation" data-cites="wdzgXUGy">[<a href="#ref-wdzgXUGy" role="doc-biblioref">38</a>]</span>. Cell types need to be represented ontologically (in the computational sense), which can be thought of multiple, intertwining trees, which take into account different ways of classifying cells.</p>
<h1 id="practical-consequences-of-the-definition">Practical consequences of the definition</h1>
<p>In the previous section, we discussed the logical entailments of accepting the proposed rules as valid. Here, we will extend the pragmatic considerations on using such a system for real-world applications.</p>
<p>In one recent attempt to define cell types for single cell RNA-Seq, Aevermann et al came up with a set of needs: “The minimum set of necessary and sufficient marker genes selectively expressed by the cell type”, “A parent cell class in the CL”, and “A specimen source description (anatomic structure þ species).” <span class="citation" data-cites="IgMqOHJU">[<a href="#ref-IgMqOHJU" role="doc-biblioref">39</a>]</span> They have great merit in defining clear guidelines for marking a cell type. The requirement of markers is reasonable for the field of single-cell RNA-seq, where marker information is abundant. The Cell Ontology has used markers for defining cell types, an approach specially employed for immune cells <span class="citation" data-cites="gAysgnsS">[<a href="#ref-gAysgnsS" role="doc-biblioref">19</a>]</span> <span class="citation" data-cites="9UTyLFor">[<a href="#ref-9UTyLFor" role="doc-biblioref">26</a>]</span> <span class="citation" data-cites="17uICvtFm">[<a href="#ref-17uICvtFm" role="doc-biblioref">27</a>]</span>.</p>
<p>The use of markers, however, leaves us with a conceptual problem: definitions of cell type used by electrophysiologists, or even in the manuals of histology classes, are not based on markers. Rigorously, this would leave aside a whole part of what we consider biomedical knowledge. Moreover, gene markers are not defined for cell types that span multiple species, a problem already discussed on the Cell Ontology report of 2011 <span class="citation" data-cites="gAysgnsS">[<a href="#ref-gAysgnsS" role="doc-biblioref">19</a>]</span>.</p>
<p>By using our less stringent set of rules, we can better evaluate claims of discover of new cell types. With vast amounts of data and loose definition of cell types, it becomes uncannily easy to claim a new cell type. Converserly, if one explictly claims to discover a new “stricto sensu” cell type, one has to provide enough evidence that cells from this class are identifiable across all individuals of a species. A claim of an “archetype” would require evidence for existence in more than one species. Consequently, experiments that only use a specific strain of mice have a stronger claim if the expectation is limited to the infratype.</p>
<p>An example of the discovery of a new “archetype” is the pair of articles published in Nature in 2018 <span class="citation" data-cites="ZZJyod8A">[<a href="#ref-ZZJyod8A" role="doc-biblioref">40</a>]</span> <span class="citation" data-cites="c7kshLTJ">[<a href="#ref-c7kshLTJ" role="doc-biblioref">41</a>]</span> about the newly found “ionocyte”, a class of cells in the trachea enriched for expression of genes homologous to the <em>CFTR</em> gene. Both studies displayed evidence for such a class in both mouse and human samples, corroborating the existence of an archetype. This discovery of an archetype has been denominated by both articles as a discovery of a new cell type.</p>
<p>Another example of cell type discovery is a pioneer article by Villani et al <span class="citation" data-cites="10Jn8geg8">[<a href="#ref-10Jn8geg8" role="doc-biblioref">42</a>]</span>. It describes subclasses of monocytes and dendritic cells in humans, and pragmatically uses markers for their definition. The patients were recruited from “the Boston-based PhenoGenetic project (…) and the Newcastle community.” It is arguable that they did not have a random sample of humanity, and the observed results might not hold for different populations. This discovery of infratypes has also been described as a discovery of a new “cell type”.</p>
<p>An example from the article is the discovery of the “AS Dendritic cell” (and two subpopulations of it), characterized by expression of the antigens for the proteins AXL and SIGLEC6. This and other cell types are presented in the article as part of a “Human dendritic cell atlas”, generalizing the theory for the whole humanity. However, it is not clear if the population sampled included individuals from different human background. Thus, it is technically possible that the existence of the “cell type” exactly as described might be restricted to some human groups. The jump from technotype (which takes into consideration also descriptors like “healthy” and “age between 25 and 40 years”) to infratype (“all humans in this population scope”) to cell type <em>strictu sensu</em>(all humans) is depicted in the figure <a href="#fig:villani">1</a> and exemplifies the logical flow.</p>
<p>“Dendritic cells” is one of the cell types most thoroughly modelled by Cell Ontology. <span class="citation" data-cites="CAzXpaob">[<a href="#ref-CAzXpaob" role="doc-biblioref">43</a>]</span> <span class="citation" data-cites="doi:10.1016/j.jbi.2010.01.006">[<span class="citeproc-not-found" data-reference-id="doi:10.1016/j.jbi.2010.01.006"><strong>???</strong></span>]</span>. The current natural language definition of dendritic cell (<a href="http://purl.obolibrary.org/obo/CL_0000451">CL_0000451</a>) states that a dendritic cell is “A cell of hematopoietic origin, typically resident in particular tissues, specialized in the uptake, processing, and transport of antigens to lymph nodes for the purpose of stimulating an immune response via T cell activation. These cells are lineage negative (CD3-negative, CD19-negative, CD34-negative, and CD56-negative).” The structured definitions are derived from the leukocyte (<a href="http://purl.obolibrary.org/obo/CL_0000738">CL_0000738</a>) definition, which defines such cells as “achromatic cell of the myeloid or lymphoid lineages capable of ameboid movement.” These definitions are not reconcilable to the “dendritic cells” studied by Villani et al’s. We have no way of knowing if the cells in their work are “typically resident in particular tissues”, “achromatic” or “capable of ameboid movement”. That might sound pedantic, and might unfortunately be, but the logical requirements of computational systems leads to both <a href="https://www.biocuration.org/community/biocuration-generic-job-description/">biocurators</a> and <a href="https://eloquentjavascript.net/00_intro.html">computers</a> being seen as pedantic. This high level of precision is necessary to accurately depict not only the complexities of cell types, but the complexities of research settings.</p>
<div id="fig:villani" class="fignos">
<figure>
<img src="images/villani_asdcs.png" alt="" /><figcaption><span>Figure 1:</span> Conceptualization of a set of the cell types in Villani et al, 2017 <span class="citation" data-cites="10Jn8geg8">[<a href="#ref-10Jn8geg8" role="doc-biblioref">42</a>]</span>. The depicted cell types were manually curated from the article, where they are either implicitly or explicitly mentioned. The set of cell types is not comprehensive, and represent only a small fraction of the concepts handled by the authors. Identifiers for cell types are written in pseudocode based on the Turtle serialization for RDF knowledge graphs (https://www.w3.org/TR/turtle/) and represent valid URIs (described in the database https://celltypes.wiki.opencura.com/wiki/Main_Page). URI: Universal Resource Identifier; RDF: Resource Description Framework.</figcaption>
</figure>
</div>
<p>Even if we are not yet able to formally represent all the aspects that go into a cell type definition, we can use an explicit “natural language definition” property to define cell types. As David Osumi-Sutherland puts in his 2017 article about cell type classification, there is a “<em>mismatch between quantified logic, which records assertions about all members of a class, and the messy, noisy reality of biology and the data we collect about it.</em>.” <span class="citation" data-cites="KJDots8y">[<a href="#ref-KJDots8y" role="doc-biblioref">28</a>]</span>. We do not need to have all of biology classified before we deal with cell types. Taking as example in figure <a href="#fig:villani">1</a>, all cell types treated as “dendritic cells” in the literature are valid subclasses of the dendritic cell archetype (ct:Q20). Such a subclassing system might lack the power to computationally check the validity of definitions. However, by the principle of minimal commitment <span class="citation" data-cites="doi:10.1.1.89.5775">[<span class="citeproc-not-found" data-reference-id="doi:10.1.1.89.5775"><strong>???</strong></span>]</span>, it could already be be a suitable scaffold for representing experimental data (e.g from single-cell transcriptomics) and allow logically robust data integration.</p>
<p>Sabina Leonelli stated that the challenges brought up by big data in biology require an advance of our philosophical theories <span class="citation" data-cites="OQp5PCry">[<a href="#ref-OQp5PCry" role="doc-biblioref">44</a>]</span>. We agree, and argue that the inverse is also true: to advance the theoretical foundations of modern biology, we need to harness the power of computational tools. Computational ontologies provide a solution for dealing with complex concepts. Classes in ontologies can have alpha-numeric identifiers. We can, thus assign each technotype a Unique Resource Identifier, a URI, similar to the Cell Ontology (CL) <span class="citation" data-cites="1CCBSogYG">[<a href="#ref-1CCBSogYG" role="doc-biblioref">18</a>]</span> <span class="citation" data-cites="gAysgnsS">[<a href="#ref-gAysgnsS" role="doc-biblioref">19</a>]</span> <span class="citation" data-cites="17U3pGEjv">[<a href="#ref-17U3pGEjv" role="doc-biblioref">20</a>]</span> or Wikidata <span class="citation" data-cites="HUodcdz1">[<a href="#ref-HUodcdz1" role="doc-biblioref">45</a>]</span> <span class="citation" data-cites="o5Fiwpr0">[<a href="#ref-o5Fiwpr0" role="doc-biblioref">46</a>]</span>. The power of using knowledge bases for integrating knowledge about cell types is becoming clear <span class="citation" data-cites="TupzTvtA">[<a href="#ref-TupzTvtA" role="doc-biblioref">13</a>]</span>, and its success relies heavily on precise usage of unique identifiers.</p>
<p>The quest for names/labels systems for types is related to the quest for defining cell types, and exciting proposals are already arising. <span class="citation" data-cites="SERJvWOl">[<a href="#ref-SERJvWOl" role="doc-biblioref">47</a>]</span> <span class="citation" data-cites="GfyYemJn">[<a href="#ref-GfyYemJn" role="doc-biblioref">48</a>]</span> Nevertheless, the definition of rules for <em>naming</em> cell types is outside the scope of this article, and abstracted away via semantically void identifiers. A semantically void identifier, in contrast to rule-based numeric identifiers<span class="citation" data-cites="doi:10.1002/cyto.a.23346">[<span class="citeproc-not-found" data-reference-id="doi:10.1002/cyto.a.23346"><strong>???</strong></span>]</span> or natural language names (such as Latin scientific names for species) is resilient to changes in our knowledge. It also floats away from the problems of essence-indicating and role-indicating names discussed by Rowe and Stone in 1977<span class="citation" data-cites="j70Pv1a">[<a href="#ref-j70Pv1a" role="doc-biblioref">49</a>]</span>. Complementary to being non-descriptive, identifiers can be assigned labels, which can be freely changed, while keeping a persistent Universal Resource Identifier. The We devoted our effort to refining the logical aspects of cell-type definitions, which might be combined with any commonly agreed naming/labeling system.</p>
<p>The level of “technotype” allows precise labeling of cell types in real world experiments. The non-existence of truly exact matches in CL (even when combined with other ontologies) renders it theoretically impossible to annotate articles and datasets with perfection. The “technotype” gives power to every researcher to craft their “cell type” of interest and connect it to a common network of knowledge.</p>
<p>Specifically, for single-cell transcriptomics, the technotype refines our model for labeling cells (and, consequently, cell clusters). A branch of computational single-cell development has dedicated itself to find tools for labeling single-cell experiments. While some approaches ignore ontologies <span class="citation" data-cites="I4hT5zr7">[<a href="#ref-I4hT5zr7" role="doc-biblioref">36</a>]</span> <span class="citation" data-cites="6nwUdsGG">[<a href="#ref-6nwUdsGG" role="doc-biblioref">50</a>]</span>, others aim at finding the best class among the Cell Ontology <span class="citation" data-cites="1H6dBIKAi">[<a href="#ref-1H6dBIKAi" role="doc-biblioref">51</a>]</span> <span class="citation" data-cites="cosf0Kp4">[<a href="#ref-cosf0Kp4" role="doc-biblioref">52</a>]</span> or MeSH IDs<span class="citation" data-cites="ixNexUPF">[<a href="#ref-ixNexUPF" role="doc-biblioref">53</a>]</span> <span class="citation" data-cites="TXlkRBJh">[<a href="#ref-TXlkRBJh" role="doc-biblioref">54</a>]</span>. Manual matches to the Cell Ontology have been fed to algorithms such as BLAST2CO <span class="citation" data-cites="TXlkRBJh">[<a href="#ref-TXlkRBJh" role="doc-biblioref">54</a>]</span> and OnClass <span class="citation" data-cites="e9KE53FC">[<a href="#ref-e9KE53FC" role="doc-biblioref">55</a>]</span> to predict best “matches” for a single-cell RNA-seq cells or clusters. However, unless the cells were sampled in the same way across articles, and drawn at random from the same population of individuals, they represent strictly different classes, even if very similar. Thus, we must change the task from finding a “match” to the cells in a given current experiment to finding a “point of insertion” in an ontological network. By acknowledging these real differences, we can have precise metadata, enabling precise statements and facilitating valid reuse of publically available data.</p>
<p>An Occan`s razor argument for the use of the technotype is that it makes data integration a simple task. Let a <em>stricto sensu</em> human type X be <em>completely</em> defined by “any cell that expresses the <em>CD4</em> gene, but not the <em>CD8A</em> gene”. A definition of this kind allows detection via both mRNA or proteins, as both are evidence that a cell is <em>expressing</em> a gene. Then, only the information about these two genes is used, and the analyst has only to decide what is the threshold for considering a gene as expressed. All cells that matches such pattern would be assigned to a technotype Y, where Y is a subclass of the <em>stricto sensu</em> human type X. Each cell could receive more than one label (depending on the masks applied to the dataset).</p>
<p>Some cells of interest could be missed, especially taking into account dropouts in single-cell. However, with a carefully crafted set of necessary markers, such features can be explicitly accounted for. A cell type definion can employ any logical relation, including “OR”. A cell type W can be defined as any cell that expresses the <em>MS4A1</em> gene OR the <em>CD79A</em> gene“. Once such a marker-based definition is decided by the researcher given their research questions, it can be easily tested for any cell, regardless of the size of the dataset. In computational terms, regarding an increasing number of features in the dataset, the mask algorithm scales in big O(1), and regarding the number of cells (provided thad you want to label all cells) in big O(n). If there is no information about the expression of”<em>MS4A1</em> gene OR the <em>CD79A</em> gene", no cell in the dataset can be assigned to <em>exactly</em> this cell type, and another classes need to be employed. Our flexible, yet rigorous, framework for defining cell types can help us to deal with the challenges of varying resolution levels of interest and the scaling large datasets. [doi:10.1186/s13059-020-1926-6]</p>
<p>Such a rule based system enables interpretable data-driven cell-type classifications, in contrast to the previously proposed cell type definitions by reference expression matrices with tens of thousands of genes. Furthermore, it makes the task of <em>identification</em> of cell types almost trivial: you only know what you measure, and that can be checked against an explicit and complete definition (containing the features that are necessary and sufficient for an identification). Any logical entailments (e.g : a cell that expresses CD3 is achromatic) must be acknowledge explicitly too, if we want to use them for classification and identification. The need of an identification routine for cell-type taxonomies is acknowledge for more than 45 years <span class="citation" data-cites="l6iqnp9w">[<a href="#ref-l6iqnp9w" role="doc-biblioref">56</a>]</span>, and still is a core challenge of human cell-type atlases [doi:10.1186/s13059-020-1926-6]. The quest for data-driven cell classification is at least as old <span class="citation" data-cites="l6iqnp9w">[<a href="#ref-l6iqnp9w" role="doc-biblioref">56</a>]</span>. The framework here proposed provides ideas to improve our solutions to both tasks.</p>
<h1 id="final-remarks">Final remarks</h1>
<p>In this article, we proposed a set of 3 rules (rigorous description, taxon scope restriction, and theoretical usefulness) and 1 recommendation (hierarchical link to other cell types) to define cell types. We also proposed 4 namings to clarify discussions on the topic: archetypes (a class with a scope above species level), <em>stricto sensu</em> cell types (a class with scope equal to one species), infratypes (a class with scope below the species level) and technotypes (the exact cell type defined for an experimental setup). The concept of the “technotype” can be harnessed as the unit for classifying cells, in an analogous way of how the “species” is the conventional unit for classifying organisms into higher-order taxa.</p>
<p>We dissected some logical entailments of such definition, which, admittedly, might conflict with current views on defining cell types. We do not aim at solving such conflicts, or negating the other perspectives, but only propose a different way of organizing our knowledge about cell types.</p>
<p>This article clarifies some of the meanings and provide directions to the future development of the theoretical basis of cell type definition. The discussion on cell types’ definition is still on its infancy, and we need human power to tackle this huge theoretical challenges. Biologists, philosophers, and computer scientists ought to distill the details of defining cell types, powering the Human Cell Atlas, and the life sciences research enterprise of this century.</p>
<h1 id="acknowledgements">Acknowledgements</h1>
<p>We would like to express our gratitude and acknowledge the researches that dedicated time specifically to help us to discuss and refine the basis for developing concepts here presented. Nominally, we thank Kleber Neves, Gabriel Lovate, Cesar Prada, Diógenes Saulo Lima, Lucas Cardozo, Juliane Fernandes, Pedro Medeiros, Érika Molina, Antonio Pedro Vieira, João Vitor Cavalcante, Maria Fernanda Forni, Diorge Souza, Jean Bezerra, Gabriel Sato, Roberta Andrejew, Dimitrius Pramio, Shunfu Mao, Anna Yannakopoulos and Stephanie Hickey for the discussions. The work was also shaped by question-answer interactions with David Osumi-Sutherland and Alexander Diehl (CELLS Workshop and Cell Ontology training 2020), Aviv Regev (HCA Latin America 2020) and Cole Trapnell (ISCB 2020). Part of this work was supported by grant #2019/26284-1, São Paulo Research Foundation (FAPESP).</p>
<h1 id="supplementary-notes-which-did-not-end-in-the-text">Supplementary notes which did not end in the text</h1>
<p>A set of notes that may be incorporated into the final text, or become appendices, or end up as blog posts somewhere.</p>
<h2 id="what-to-do-when-two-researchers-disagree-on-a-definition">What to do when two researchers disagree on a definition?</h2>
<p>Cell type names are notoriously ambiguous and one definition might collide with an other, specially regarding the natural language name used to described. There are many different, equally valid definitions of a “dendritic cell.” we do not aim to solve this problem from a societal standpoint. However, from a computational-ontology standpoint, there is one simple solution: split the concept.</p>
<p>This approach is similar to King Solomon’s solution in a famous bible story, called the <a href="https://en.wikipedia.org/wiki/Judgment_of_Solomon">Judgement of Solomon</a>. In a dispute between two women that claimed to be the mothers of a child, the solution of the king was simple: split the baby. However, babies are notoriously indivisible, and the true mother did not really like the idea.</p>
<p>It may be that some scientists are attached to their names, as mothers are to their babies. However, unlike babies, namings can be divided. Each of the scientists gets to name their specific conceptualization however they choose. Many names might “collide” in that way, and that is okay. Under the hood, however, the names refer to different identifiers. Computationally there would be no ambiguity. Then, it is just a matter of the researcher to respect the choice of their peers of calling something by the <em>wrong</em> name, as long as the identifier is correct.</p>
<p>Splitting concepts upon conflicts in the end is more the multiplication of bread and fish in the <a href="https://en.wikipedia.org/wiki/Feeding_the_multitude">Feeding the multitude</a> episode, and everyone gets to eat.</p>
<p>But ontologies are different from ordinary babies and magical fish. The splitting of concepts would not only create new concepts, but leave a trace. They would be immediate subclasses of their conjunction. An equally valid superclass that can be defined by “a cell containing characteristics of any of their subclasses”.</p>
<p>In a parallel with text-book mitosis, the concept gets divided in two new, equally real concepts. And as we can trace cells in an animal to a single zygote, we can keep track of concepts while they keep dividing, whenever a new conflict pops up.</p>
<h2 id="the-big-assumption-of-continuity-in-time">The big assumption of continuity in time</h2>
<p>One assumption that underlies the validity of the models proposed here is that taxons preserve their characteristics throughout time.</p>
<p>In Popper’s Logic of Scientific Research, he states that he has a metaphysical faith on the continuity of laws of nature through time.</p>
<p>We have no way of testing this metaphysical faith, and it is absolutely necessary for the scientific endeavour as we understand.</p>
<p>While in physics this assumption seems to be reasonable, evolution makes biology quite more complicated. Statements that we have about the human species, for example, might be valid today, but were not valid 2000 years ago, and vice-versa.</p>
<p>When we talk about sub-species taxons, which might be a local population of a town, for example, this unit is not immutable. The population of Newcastle, as per the example, might change in time, with imigration and emigration, mutation, natural selection, neutral evolution and the many forms of modifications of a gene pool.</p>
<p>It is, thus, and heuristic, to call “population of Newcastle” a class. We could specify a period in time for which we expect the information to be valid. For example, we may say we are sampling from “the Newcastle population in the years 2019-2020.” This would be a valid statement, but it would not be falsifiable, as by 1st of January 2021, no independent tests of the theory can be done.</p>
<p>It is technically possible to have a technotype so precise as to have a scope with a time constraint. In fact, that might be the right way of representing information, if we want to compare experiments done in evolving populations.</p>
<p>While evolutionary definitions take this dimension into account, they are fit to theoretical research, but still lack the rigour for explaining real world experiments.</p>
<p>All research that uses human samples are subject to strong influence of time.</p>
<p>Thus, the explicit assumption here is that taxons are consistent in time. And, for what we know, it is blatantly false.</p>
<p>This is a great flaw of the model, and, maybe, of a great part of biomedical research. The logical consequences are so dire, that it merits a separate, dedicated work.</p>
<h2 id="clusters-are-not-cells">Clusters are not cells</h2>
<p>In the era of large-scale omics, we are starting to see declarations of cell types that are not based on pre-selected criteria, but derived from unsupervised clustering followed by labelling.</p>
<p>This is a powerful exploratory approach, which, as mentioned in the main text, has led to discoveries of ionocytes and new classes of dendritic cells, for example.</p>
<p>Many algorithms and “expert-based” annotation protocols focus on labeling <em>clusters</em> instead of labeling <em>cells</em>.</p>
<p>Cells in a cluster are arbitrarily similar (as determined by the clustering algorithm) and so they will, by definition, differ from other cells in the sample.</p>
<p>For single-cell RNA-seq, one usually checks which genes are differentially expressed when comparing the cells in a cluster with cells in other clusters. These genes are called “markers” and used for labeling a cell cluster.</p>
<p>What does it mean to label a cell cluster, though? Does it mean that <em>all</em> cells there conform to the cell type? Does it mean that <em>most</em> cells there conform to the cell type? Does it mean that cells from other clusters in the dataset <em>definitely do not</em> conform to the cell type?</p>
<p>So far, we haven’t seem a clear, explicit, coherent definition for a cluster label. Not even once.</p>
<p>Marker-based definitions are assumed for the group as a whole, but in current pipelines, nothing blocks one cell in a cluster to lack the expression of a “name-giver” marker.</p>
<p>The classification scheme proposed here works to classify <em>cells</em>, but is not sufficient for labeling unsupervisedly-defined <em>cell clusters</em>.</p>
<p>What is possible, though, is to use clustering for data exploration. From then on, strict patterns can be decided (ex: a cell that expresses A and B, but not C) and then apply this pattern to the whole dataset. For clusters with consistent markers, this approach should be roughly equivalent to the previously described.</p>
<p>Using such “regular expressions” might lead to a cell being assigned to multiple clusters. Even if we assume that the sample is free of doublets, that cannot be a problem. Cells may have multiple functions. As argued in the main text, each cell can be labeled by multiple standards.</p>
<p>We may avoid multiple labeling if we really need in practice, though, and make preferential claims (if a cell matches definitions X and Y, it is assigned only to X, for example).</p>
<p>By having explicit “regular expression” patterns for cell definitions in single-cell datasets, the “cell-type assessment” problem becomes trivial: a cell in a new dataset is of the <em>exact</em> same type if (and only if) it matches the <em>exact</em> definition.</p>
<p>When that is not the case, current algorithms for reconciling single cell datasets can still be successfully employed. But instead of propagating a label, it would propagate a parent class, looking for cells of a similar, sister class.</p>
<h2 id="what-this-work-is-not">What this work is not</h2>
<p>This is not an attempt to substitute the Cell Ontology (CL) or contradict it in any way. CL is an amazing resource, built by a community of wonderful researchers. Its relation with CL is coexistential, and topics discussed here might be or might not be of interest of CL, and that is OK.</p>
<p>This is not an attempt to create an ontology itself, or a system that allows reasoning. It is a set of suggestions that can be taken into consideration for building a coherent ontology. The <a href="https://celltypes.wiki.opencura.com/">Cell Type Wikibase</a> is an experimental ontology, and far from ready for professional use.</p>
<p>This is not an attempt to have a one-size-fits-all definition of cell type. It is built as a theoretical solution for one cell-type related task. Similar to species definitions, we need an ecosystem of cell-type and cell state definitions that better suit different areas.</p>
<p>This is not an attempt to claim anything about the “true” nature of cell types, in the biological sense. It is a proposal of pratical guidelines to represent research data.</p>
<p>This is not an attempt to solve <em>all</em> problems for cell type data annotation. It is the introduction of alternatives that need to be further developed and discussed.</p>
<h2 id="immune-epithope-db">Immune Epithope DB</h2>
<p>The Immune Epitope Database and Analysis Resource (IEDB) announced it in 2006 [doi:10.1371/journal.pcbi.0020125]:</p>
<p>“the goal of the IEDB is to present as much information as possible without subjective interpretation, we can never presume any information, but rather we must try to capture the data exactly as presented in the reference, while maintaining the conclusions of the reference in a uniform manner. For example, if all experiments are performed with a whole cell population, but the authors attribute the response to a particular cell type without any evidence, we must capture the effector cells as the entire population.”</p>
<p>That is the kind of challenge that the “technotype” solves in theory, as it gets the objective population sampled in any article.</p>
<h2 id="phenetic-species-concept">Phenetic Species Concept</h2>
<p>Thn Phenetic Species Concept states that species are the smallest groups that are consistently and persistently distinct and distinguishable by ordinary means.
Maybe that is crosslinkable.</p>
<p>The link between the phenetic species concept and the classification of cell types has been noted before. <span class="citation" data-cites="klRLs2Ss">[<a href="#ref-klRLs2Ss" role="doc-biblioref">57</a>]</span>
The ideas of explicit definitions and hierarchical relations resound with the goals of this article.</p>
<p>In this work<span class="citation" data-cites="klRLs2Ss">[<a href="#ref-klRLs2Ss" role="doc-biblioref">57</a>]</span> , they mention 5 facets of phenetic systematics:
- Group (that is, ‘type’) membership should be based on multiple criteria rather than on a single so‑called ‘essential’ feature that the investigator favours.
- The criteria for group membership should be rule‑based, explicit and quantitative.
- Groupings should be hierarchical rather than flat to acknowledge the validity of both coarse and fine divisions.
- Groupings generated by this approach should be viewed as hypotheses to be tested rather than inflexible rules.
- Classification should focus on discontinuities between groups and ignore parameters that vary continuously.</p>
<h2 id="calculation-of-theoretical-maxima-and-minima-of-human-cell-types">Calculation of theoretical maxima and minima of human cell types</h2>
<p>https://twitter.com/lubianat/status/1295923945770823682</p>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references hanging-indent" role="doc-bibliography">
<div id="ref-E78joy5H">
<p>1. <strong>The structure of scientific revolutions</strong> <br />
Thomas S. Kuhn, Ian Hacking<br />
<em>The University of Chicago Press</em> (2012) <br />
ISBN: <a href="https://worldcat.org/isbn/9780226458113">9780226458113</a></p>
</div>
<div id="ref-H4OiWiM2">
<p>2. <strong>Neuronal cell types</strong> <br />
Richard H Masland<br />
<em>Current Biology</em> (2004-07) <a href="https://doi.org/dg84br">https://doi.org/dg84br</a> <br />
DOI: <a href="https://doi.org/10.1016/j.cub.2004.06.035">10.1016/j.cub.2004.06.035</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/15242626">15242626</a></p>
</div>
<div id="ref-jePJgR0H">
<p>3. <strong>Search BioNumbers - The Database of Useful Biological Numbers</strong> <a href="https://bionumbers.hms.harvard.edu/search.aspx">https://bionumbers.hms.harvard.edu/search.aspx</a></p>
</div>
<div id="ref-V85zvNGa">
<p>4. <strong>How Many Types of Cells Are in the Human Body?</strong> <br />
ibswit<br />
(2017-05-17) <a href="https://askabiologist.asu.edu/questions/human-cell-types">https://askabiologist.asu.edu/questions/human-cell-types</a></p>
</div>
<div id="ref-H4wZSl5d">
<p>5. <strong>What Is Your Conceptual Definition of “Cell Type” in the Context of a Mature Organism?</strong> <br />
Cell Systems<br />
(2017-03) <a href="https://doi.org/d38b">https://doi.org/d38b</a> <br />
DOI: <a href="https://doi.org/10.1016/j.cels.2017.03.006">10.1016/j.cels.2017.03.006</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28334573">28334573</a></p>
</div>
<div id="ref-vk9ZInF3">
<p>6. <strong>The Human Cell Atlas</strong> <br />
Aviv Regev, Sarah A Teichmann, Eric S Lander, Ido Amit, Christophe Benoist, Ewan Birney, Bernd Bodenmiller, Peter Campbell, Piero Carninci, Menna Clatworthy, … Human Cell Atlas Meeting Participants<br />
<em>eLife</em> (2017-12-05) <a href="https://doi.org/gcnzcv">https://doi.org/gcnzcv</a> <br />
DOI: <a href="https://doi.org/10.7554/elife.27041">10.7554/elife.27041</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29206104">29206104</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5762154">PMC5762154</a></p>
</div>
<div id="ref-167nEKPOQ">
<p>7. <strong>The human body at cellular resolution: the NIH Human Biomolecular Atlas Program.</strong><em>Nature</em> (2019-10-09) <a href="https://www.ncbi.nlm.nih.gov/pubmed/31597973">https://www.ncbi.nlm.nih.gov/pubmed/31597973</a> <br />
DOI: <a href="https://doi.org/10.1038/s41586-019-1629-x">10.1038/s41586-019-1629-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31597973">31597973</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6800388">PMC6800388</a></p>
</div>
<div id="ref-5QxCvchL">
<p>8. <strong>What Is Your Conceptual Definition of “Cell Type” in the Context of a Mature Organism?</strong> <br />
Cell systems<br />
(2017-03-22) <a href="https://www.ncbi.nlm.nih.gov/pubmed/28334573">https://www.ncbi.nlm.nih.gov/pubmed/28334573</a> <br />
DOI: <a href="https://doi.org/10.1016/j.cels.2017.03.006">10.1016/j.cels.2017.03.006</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28334573">28334573</a></p>
</div>
<div id="ref-sWv3okaH">
<p>9. <strong>A periodic table of cell types</strong> <br />
Bo Xia, Itai Yanai<br />
<em>Development</em> (2019-06-15) <a href="https://doi.org/ggctwf">https://doi.org/ggctwf</a> <br />
DOI: <a href="https://doi.org/10.1242/dev.169854">10.1242/dev.169854</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31249003">31249003</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6602355">PMC6602355</a></p>
</div>
<div id="ref-19yYJWEkz">
<p>10. <strong>Exciting times to study the identity and evolution of cell types</strong> <br />
Maria Sachkova, Pawel Burkhardt<br />
<em>Development</em> (2019-09-19) <a href="https://doi.org/ghdb9v">https://doi.org/ghdb9v</a> <br />
DOI: <a href="https://doi.org/10.1242/dev.178996">10.1242/dev.178996</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31537583">31537583</a></p>
</div>
<div id="ref-1DYAsd4eB">
<p>11. <strong>The Human Cell Atlas: from vision to reality</strong> <br />
Orit Rozenblatt-Rosen, Michael J. T. Stubbington, Aviv Regev, Sarah A. Teichmann<br />
<em>Nature</em> (2017-10-26) <a href="https://doi.org/gfgkr8">https://doi.org/gfgkr8</a> <br />
DOI: <a href="https://doi.org/10.1038/550451a">10.1038/550451a</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29072289">29072289</a></p>
</div>
<div id="ref-7zWO5k6T">
<p>12. <strong>Human Cell Atlas and cell-type authentication for regenerative medicine</strong> <br />
Yulia Panina, Peter Karagiannis, Andreas Kurtz, Glyn N. Stacey, Wataru Fujibuchi<br />
<em>Experimental &amp; Molecular Medicine</em> (2020-09) <a href="https://doi.org/ghfckj">https://doi.org/ghfckj</a> <br />
DOI: <a href="https://doi.org/10.1038/s12276-020-0421-1">10.1038/s12276-020-0421-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32929224">32929224</a></p>
</div>
<div id="ref-TupzTvtA">
<p>13. <strong>A community-based transcriptomics classification and nomenclature of neocortical cell types</strong> <br />
Rafael Yuste, Michael J. Hawrylycz, Nadia Aalling, Argel Aguilar-Valles, Detlev Arendt, Rubén Armañanzas, Giorgio A. Ascoli, Concha Bielza, Vahid Bokharaie, Tobias B. Bergmann, … Ed S. Lein<br />
<em>Nature Neuroscience</em> (2020-08-24) <a href="https://www.wikidata.org/wiki/Q98665291">https://www.wikidata.org/wiki/Q98665291</a> <br />
DOI: <a href="https://doi.org/10.1038/s41593-020-0685-8">10.1038/s41593-020-0685-8</a></p>
</div>
<div id="ref-w0Woz7JE">
<p>14. <strong>The evolving concept of cell identity in the single cell era</strong> <br />
Samantha A. Morris<br />
<em>Development</em> (2019-06-27) <a href="https://www.wikidata.org/wiki/Q93086971">https://www.wikidata.org/wiki/Q93086971</a> <br />
DOI: <a href="https://doi.org/10.1242/dev.169748">10.1242/dev.169748</a></p>
</div>
<div id="ref-O95WN4x0">
<p>15. <strong>The evolution of cell types in animals: emerging principles from molecular studies.</strong> <br />
Detlev Arendt<br />
<em>Nature reviews. Genetics</em> (2008-11) <a href="https://www.ncbi.nlm.nih.gov/pubmed/18927580">https://www.ncbi.nlm.nih.gov/pubmed/18927580</a> <br />
DOI: <a href="https://doi.org/10.1038/nrg2416">10.1038/nrg2416</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18927580">18927580</a></p>
</div>
<div id="ref-ogAGO9KH">
<p>16. <strong>The origin and evolution of cell types</strong> <br />
Detlev Arendt, Jacob M. Musser, Clare V. H. Baker, Aviv Bergman, Connie Cepko, Douglas H. Erwin, Mihaela Pavlicev, Gerhard Schlosser, Stefanie Widder, Manfred D. Laubichler, Günter P. Wagner<br />
<em>Nature Reviews Genetics</em> (2016-11-07) <a href="https://doi.org/f9b62x">https://doi.org/f9b62x</a> <br />
DOI: <a href="https://doi.org/10.1038/nrg.2016.127">10.1038/nrg.2016.127</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27818507">27818507</a></p>
</div>
<div id="ref-g74dZjOM">
<p>17. <strong>Species Concepts and Species Delimitation</strong> <br />
Kevin De Queiroz<br />
<em>Systematic Biology</em> (2007-12) <a href="https://doi.org/c34kzf">https://doi.org/c34kzf</a> <br />
DOI: <a href="https://doi.org/10.1080/10635150701701083">10.1080/10635150701701083</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18027281">18027281</a></p>
</div>
<div id="ref-1CCBSogYG">
<p>18.<strong>:{unav)</strong> <br />
Jonathan Bard, Seung Y Rhee, Michael Ashburner<br />
<em>Genome Biology</em> (2005) <a href="https://doi.org/dfxc74">https://doi.org/dfxc74</a> <br />
DOI: <a href="https://doi.org/10.1186/gb-2005-6-2-r21">10.1186/gb-2005-6-2-r21</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/15693950">15693950</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC551541">PMC551541</a></p>
</div>
<div id="ref-gAysgnsS">
<p>19. <strong>Logical Development of the Cell Ontology</strong> <br />
Terrence F Meehan, Anna Maria Masci, Amina Abdulla, Lindsay G Cowell, Judith A Blake, Christopher J Mungall, Alexander D Diehl<br />
<em>BMC Bioinformatics</em> (2011-01-05) <a href="https://doi.org/c7kw6x">https://doi.org/c7kw6x</a> <br />
DOI: <a href="https://doi.org/10.1186/1471-2105-12-6">10.1186/1471-2105-12-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21208450">21208450</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3024222">PMC3024222</a></p>
</div>
<div id="ref-17U3pGEjv">
<p>20. <strong>The Cell Ontology 2016: enhanced content, modularization, and ontology interoperability</strong> <br />
Alexander D. Diehl, Terrence F. Meehan, Yvonne M. Bradford, Matthew H. Brush, Wasila M. Dahdul, David S. Dougall, Yongqun He, David Osumi-Sutherland, Alan Ruttenberg, Sirarat Sarntivijai, … Christopher J. Mungall<br />
<em>Journal of Biomedical Semantics</em> (2016-07-04) <a href="https://doi.org/gg99b9">https://doi.org/gg99b9</a> <br />
DOI: <a href="https://doi.org/10.1186/s13326-016-0088-7">10.1186/s13326-016-0088-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27377652">27377652</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4932724">PMC4932724</a></p>
</div>
<div id="ref-rK8KD0us">
<p>21. <strong>Cells in ExperimentaL Life Sciences (CELLS-2018): capturing the knowledge of normal and diseased cells with ontologies</strong> <br />
Sirarat Sarntivijai, Yongqun He, Alexander D. Diehl<br />
<em>BMC Bioinformatics</em> (2019-04-25) <a href="https://doi.org/gg99b8">https://doi.org/gg99b8</a> <br />
DOI: <a href="https://doi.org/10.1186/s12859-019-2721-9">10.1186/s12859-019-2721-9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31272374">31272374</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6509796">PMC6509796</a></p>
</div>
<div id="ref-EDf5c4SG">
<p>22. <strong>Heraclitus</strong> <br />
Daniel W. Graham<br />
<em>The Stanford Encyclopedia of Philosophy</em> (2019) <a href="https://plato.stanford.edu/archives/fall2019/entries/heraclitus/">https://plato.stanford.edu/archives/fall2019/entries/heraclitus/</a></p>
</div>
<div id="ref-cJ3Q82kT">
<p>23. <strong>Current best practices in single‐cell RNA‐seq analysis: a tutorial</strong> <br />
Malte D Luecken, Fabian J Theis<br />
<em>Molecular Systems Biology</em> (2019-06-19) <a href="https://doi.org/gf4f4d">https://doi.org/gf4f4d</a> <br />
DOI: <a href="https://doi.org/10.15252/msb.20188746">10.15252/msb.20188746</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31217225">31217225</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6582955">PMC6582955</a></p>
</div>
<div id="ref-131Mt6Qd2">
<p>24. <strong>The meanings of “function” in biology and the problematic case of de novo gene emergence</strong> <br />
Diane Marie Keeling, Patricia Garza, Charisse Michelle Nartey, Anne-Ruxandra Carvunis<br />
<em>eLife</em> (2019-11-01) <a href="https://doi.org/ggjnmv">https://doi.org/ggjnmv</a> <br />
DOI: <a href="https://doi.org/10.7554/elife.47014">10.7554/elife.47014</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31674305">31674305</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6824840">PMC6824840</a></p>
</div>
<div id="ref-56biRUeN">
<p>25. <a href="https://www.ietf.org/rfc/rfc2119.html">https://www.ietf.org/rfc/rfc2119.html</a></p>
</div>
<div id="ref-9UTyLFor">
<p>26. <strong>Reporting and connecting cell type names and gating definitions through ontologies</strong> <br />
James A. Overton, Randi Vita, Patrick Dunn, Julie G. Burel, Syed Ahmad Chan Bukhari, Kei-Hoi Cheung, Steven H. Kleinstein, Alexander D. Diehl, Bjoern Peters<br />
<em>BMC Bioinformatics</em> (2019-04-25) <a href="https://doi.org/ghbk9r">https://doi.org/ghbk9r</a> <br />
DOI: <a href="https://doi.org/10.1186/s12859-019-2725-5">10.1186/s12859-019-2725-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31272390">31272390</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6509839">PMC6509839</a></p>
</div>
<div id="ref-17uICvtFm">
<p>27. <strong>flowCL: ontology-based cell population labelling in flow cytometry</strong> <br />
Mélanie Courtot, Justin Meskas, Alexander D. Diehl, Radina Droumeva, Raphael Gottardo, Adrin Jalali, Mohammad Jafar Taghiyar, Holden T. Maecker, J. Philip McCoy, Alan Ruttenberg, … Ryan R. Brinkman<br />
<em>Bioinformatics</em> (2015-04-15) <a href="https://doi.org/f7cc46">https://doi.org/f7cc46</a> <br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/btu807">10.1093/bioinformatics/btu807</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25481008">25481008</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4393520">PMC4393520</a></p>
</div>
<div id="ref-KJDots8y">
<p>28. <strong>Cell ontology in an age of data-driven cell classification</strong> <br />
David Osumi-Sutherland<br />
<em>BMC Bioinformatics</em> (2017-12-21) <a href="https://doi.org/ghcbdk">https://doi.org/ghcbdk</a> <br />
DOI: <a href="https://doi.org/10.1186/s12859-017-1980-6">10.1186/s12859-017-1980-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29322914">29322914</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5763290">PMC5763290</a></p>
</div>
<div id="ref-wpSlfce3">
<p>29. <strong>Summary for Policymakers</strong> <br />
Cambridge University Press<br />
(2014-06-09) <a href="https://doi.org/bwnm">https://doi.org/bwnm</a> <br />
DOI: <a href="https://doi.org/10.1017/cbo9781107415324.004">10.1017/cbo9781107415324.004</a></p>
</div>
<div id="ref-xDF1wKqQ">
<p>30. <strong><em>PhyloCode</em></strong> <br />
Wikipedia<br />
(2020-11-09) <a href="https://en.wikipedia.org/w/index.php?title=PhyloCode&amp;oldid=987870644">https://en.wikipedia.org/w/index.php?title=PhyloCode&amp;oldid=987870644</a></p>
</div>
<div id="ref-1ER6m3I7b">
<p>31. <strong>PhyloCode: Division I. Principles</strong> <a href="http://phylonames.org/code/divisions/1/">http://phylonames.org/code/divisions/1/</a></p>
</div>
<div id="ref-eM5m3jNA">
<p>32. <strong>000664 - C57BL/6J</strong> <a href="https://www.jax.org/strain/000664">https://www.jax.org/strain/000664</a></p>
</div>
<div id="ref-b2FgShV8">
<p>33. <strong>The Generalizability Crisis</strong> <br />
Tal Yarkoni<br />
<em>Center for Open Science</em> (2019-11-22) <a href="https://doi.org/ggdf7h">https://doi.org/ggdf7h</a> <br />
DOI: <a href="https://doi.org/10.31234/osf.io/jqw35">10.31234/osf.io/jqw35</a></p>
</div>
<div id="ref-sFrgW0cH">
<p>34. <strong>A lógica da pesquisa científica</strong> <br />
K. R. Popper<br />
(1972) <br />
ISBN: <a href="https://worldcat.org/isbn/9788531612503">9788531612503</a></p>
</div>
<div id="ref-3H80HarF">
<p>35. <strong>Identifying gene expression programs of cell-type identity and cellular activity with single-cell RNA-Seq</strong> <br />
Dylan Kotliar, Adrian Veres, M Aurel Nagy, Shervin Tabrizi, Eran Hodis, Douglas A Melton, Pardis C Sabeti<br />
<em>eLife</em> (2019-07-08) <a href="https://doi.org/gf4q46">https://doi.org/gf4q46</a> <br />
DOI: <a href="https://doi.org/10.7554/elife.43803">10.7554/elife.43803</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31282856">31282856</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6639075">PMC6639075</a></p>
</div>
<div id="ref-I4hT5zr7">
<p>36. <strong>CHETAH: a selective, hierarchical cell type identification method for single-cell RNA sequencing</strong> <br />
Jurrian K de Kanter, Philip Lijnzaad, Tito Candelli, Thanasis Margaritis, Frank CP Holstege<br />
<em>Nucleic Acids Research</em> (2019-09-19) <a href="https://doi.org/gg99dp">https://doi.org/gg99dp</a> <br />
DOI: <a href="https://doi.org/10.1093/nar/gkz543">10.1093/nar/gkz543</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31226206">31226206</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6895264">PMC6895264</a></p>
</div>
<div id="ref-HixFZJ8X">
<p>37. <strong>Mapping the transcriptional diversity of genetically and anatomically defined cell populations in the mouse brain</strong> <br />
Ken Sugino, Erin Clark, Anton Schulmann, Yasuyuki Shima, Lihua Wang, David L Hunt, Bryan M Hooks, Dimitri Tränkner, Jayaram Chandrashekar, Serge Picard, … Sacha B Nelson<br />
<em>eLife</em> (2019-04-12) <a href="https://doi.org/ghbc3p">https://doi.org/ghbc3p</a> <br />
DOI: <a href="https://doi.org/10.7554/elife.38619">10.7554/elife.38619</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30977723">30977723</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6499542">PMC6499542</a></p>
</div>
<div id="ref-wdzgXUGy">
<p>38. <strong>How Single-Cell Genomics Is Changing Evolutionary and Developmental Biology</strong> <br />
John C. Marioni, Detlev Arendt<br />
<em>Annual Review of Cell and Developmental Biology</em> (2017-10-06) <a href="https://doi.org/ggb632">https://doi.org/ggb632</a> <br />
DOI: <a href="https://doi.org/10.1146/annurev-cellbio-100616-060818">10.1146/annurev-cellbio-100616-060818</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28813177">28813177</a></p>
</div>
<div id="ref-IgMqOHJU">
<p>39. <strong>Cell type discovery using single-cell transcriptomics: implications for ontological representation.</strong> <br />
Brian D Aevermann, Mark Novotny, Trygve Bakken, Jeremy A Miller, Alexander D Diehl, David Osumi-Sutherland, Roger S Lasken, Ed S Lein, Richard H Scheuermann<br />
<em>Human molecular genetics</em> (2018-05-01) <a href="https://www.ncbi.nlm.nih.gov/pubmed/29590361">https://www.ncbi.nlm.nih.gov/pubmed/29590361</a> <br />
DOI: <a href="https://doi.org/10.1093/hmg/ddy100">10.1093/hmg/ddy100</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29590361">29590361</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5946857">PMC5946857</a></p>
</div>
<div id="ref-ZZJyod8A">
<p>40. <strong>A revised airway epithelial hierarchy includes CFTR-expressing ionocytes</strong> <br />
Daniel T. Montoro, Adam L. Haber, Moshe Biton, Vladimir Vinarsky, Brian Lin, Susan E. Birket, Feng Yuan, Sijia Chen, Hui Min Leung, Jorge Villoria, … Jayaraj Rajagopal<br />
<em>Nature</em> (2018-08-01) <a href="https://doi.org/gdwskh">https://doi.org/gdwskh</a> <br />
DOI: <a href="https://doi.org/10.1038/s41586-018-0393-7">10.1038/s41586-018-0393-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30069044">30069044</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6295155">PMC6295155</a></p>
</div>
<div id="ref-c7kshLTJ">
<p>41. <strong>A single-cell atlas of the airway epithelium reveals the CFTR-rich pulmonary ionocyte</strong> <br />
Lindsey W. Plasschaert, Rapolas Žilionis, Rayman Choo-Wing, Virginia Savova, Judith Knehr, Guglielmo Roma, Allon M. Klein, Aron B. Jaffe<br />
<em>Nature</em> (2018-08-01) <a href="https://doi.org/gdwsj7">https://doi.org/gdwsj7</a> <br />
DOI: <a href="https://doi.org/10.1038/s41586-018-0394-6">10.1038/s41586-018-0394-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30069046">30069046</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6108322">PMC6108322</a></p>
</div>
<div id="ref-10Jn8geg8">
<p>42. <strong>Single-cell RNA-seq reveals new types of human blood dendritic cells, monocytes, and progenitors</strong> <br />
Alexandra-Chloé Villani, Rahul Satija, Gary Reynolds, Siranush Sarkizova, Karthik Shekhar, James Fletcher, Morgane Griesbeck, Andrew Butler, Shiwei Zheng, Suzan Lazo, … Nir Hacohen<br />
<em>Science</em> (2017-04-20) <a href="https://doi.org/f94x5t">https://doi.org/f94x5t</a> <br />
DOI: <a href="https://doi.org/10.1126/science.aah4573">10.1126/science.aah4573</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28428369">28428369</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5775029">PMC5775029</a></p>
</div>
<div id="ref-CAzXpaob">
<p>43. <strong>An improved ontological representation of dendritic cells as a paradigm for all cell types</strong> <br />
Anna Masci, Cecilia N Arighi, Alexander D Diehl, Anne E Lieberman, Chris Mungall, Richard H Scheuermann, Barry Smith, Lindsay G Cowell<br />
<em>BMC Bioinformatics</em> (2009) <a href="https://doi.org/cpxdhs">https://doi.org/cpxdhs</a> <br />
DOI: <a href="https://doi.org/10.1186/1471-2105-10-70">10.1186/1471-2105-10-70</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19243617">19243617</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2662812">PMC2662812</a></p>
</div>
<div id="ref-OQp5PCry">
<p>44. <strong>The challenges of big data biology</strong> <br />
Sabina Leonelli<br />
<em>eLife</em> (2019-04-05) <a href="https://doi.org/gfzw8q">https://doi.org/gfzw8q</a> <br />
DOI: <a href="https://doi.org/10.7554/elife.47381">10.7554/elife.47381</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30950793">30950793</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6450665">PMC6450665</a></p>
</div>
<div id="ref-HUodcdz1">
<p>45. <strong>Wikidata as a knowledge graph for the life sciences</strong> <br />
Andra Waagmeester, Gregory Stupp, Sebastian Burgstaller-Muehlbacher, Benjamin M Good, Malachi Griffith, Obi L Griffith, Kristina Hanspers, Henning Hermjakob, Toby S Hudson, Kevin Hybiske, … Andrew I Su<br />
<em>eLife</em> (2020-03-17) <a href="https://doi.org/ggqqc6">https://doi.org/ggqqc6</a> <br />
DOI: <a href="https://doi.org/10.7554/elife.52614">10.7554/elife.52614</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32180547">32180547</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7077981">PMC7077981</a></p>
</div>
<div id="ref-o5Fiwpr0">
<p>46. <strong>Wikidata</strong> <a href="https://www.wikidata.org/wiki/Wikidata:Main_Page">https://www.wikidata.org/wiki/Wikidata:Main_Page</a></p>
</div>
<div id="ref-SERJvWOl">
<p>47. <strong>Common Cell type Nomenclature for the mammalian brain: A systematic, extensible convention</strong> <br />
Jeremy A. Miller, Nathan W. Gouwens, Bosiljka Tasic, Forrest Collman, Cindy T. J. van Velthoven, Trygve E. Bakken, Michael J. Hawrylycz, Hongkui Zeng, Ed S. Lein, Amy Bernard<br />
<em>arXiv:2006.05406 [q-bio]</em> (2020-11-13) <a href="http://arxiv.org/abs/2006.05406">http://arxiv.org/abs/2006.05406</a></p>
</div>
<div id="ref-GfyYemJn">
<p>48. <strong>Neuron Names: A Gene- and Property-Based Name Format, With Special Reference to Cortical Neurons</strong> <br />
Gordon M. Shepherd, Luis Marenco, Michael L. Hines, Michele Migliore, Robert A. McDougal, Nicholas T. Carnevale, Adam J. H. Newton, Monique Surles-Zeigler, Giorgio A. Ascoli<br />
<em>Frontiers in Neuroanatomy</em> (2019-03-21) <a href="https://doi.org/ghf8jd">https://doi.org/ghf8jd</a> <br />
DOI: <a href="https://doi.org/10.3389/fnana.2019.00025">10.3389/fnana.2019.00025</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30949034">30949034</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6437103">PMC6437103</a></p>
</div>
<div id="ref-j70Pv1a">
<p>49. <strong>Naming of neurones. Classification and naming of cat retinal ganglion cells.</strong> <br />
Rowe MH, Stone J<br />
<em>Brain, Behavior and Evolution</em> (1977-01-01) <a href="https://www.wikidata.org/wiki/Q41052480">https://www.wikidata.org/wiki/Q41052480</a> <br />
DOI: <a href="https://doi.org/10.1159/000125660">10.1159/000125660</a></p>
</div>
<div id="ref-6nwUdsGG">
<p>50. <strong>Probabilistic gene expression signatures identify cell-types from single cell RNA-seq data</strong> <br />
Isabella N. Grabski, Rafael A. Irizarry<br />
<em>Cold Spring Harbor Laboratory</em> (2020-01-23) <a href="https://doi.org/gg99dq">https://doi.org/gg99dq</a> <br />
DOI: <a href="https://doi.org/10.1101/2020.01.05.895441">10.1101/2020.01.05.895441</a></p>
</div>
<div id="ref-1H6dBIKAi">
<p>51. <strong>ontoProc: Ontology interfaces for Bioconductor, with focus on cell type identification</strong> <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/ontoProc/inst/doc/ontoProc.html#conceptual-overview-of-ontology-with-cell-types">https://www.bioconductor.org/packages/release/bioc/vignettes/ontoProc/inst/doc/ontoProc.html#conceptual-overview-of-ontology-with-cell-types</a></p>
</div>
<div id="ref-cosf0Kp4">
<p>52. <strong>CellO: Comprehensive and hierarchical cell type classification of human cells with the Cell Ontology</strong> <br />
Matthew N. Bernstein, Zhongjie Ma, Michael Gleicher, Colin N. Dewey<br />
<em>Cold Spring Harbor Laboratory</em> (2020-07-10) <a href="https://doi.org/ghf5dt">https://doi.org/ghf5dt</a> <br />
DOI: <a href="https://doi.org/10.1101/634097">10.1101/634097</a></p>
</div>
<div id="ref-ixNexUPF">
<p>53. <strong>CellMeSH: Probabilistic Cell-Type Identification Using Indexed Literature</strong> <br />
Shunfu Mao, Yue Zhang, Georg Seelig, Sreeram Kannan<br />
<em>Cold Spring Harbor Laboratory</em> (2020-05-31) <a href="https://doi.org/gg99dr">https://doi.org/gg99dr</a> <br />
DOI: <a href="https://doi.org/10.1101/2020.05.29.124743">10.1101/2020.05.29.124743</a></p>
</div>
<div id="ref-TXlkRBJh">
<p>54. <strong>Searching large-scale scRNA-seq databases via unbiased cell embedding with Cell BLAST</strong> <br />
Zhi-Jie Cao, Lin Wei, Shen Lu, De-Chang Yang, Ge Gao<br />
<em>Nature Communications</em> (2020-07-10) <a href="https://doi.org/gg4mm3">https://doi.org/gg4mm3</a> <br />
DOI: <a href="https://doi.org/10.1038/s41467-020-17281-7">10.1038/s41467-020-17281-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32651388">32651388</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7351785">PMC7351785</a></p>
</div>
<div id="ref-e9KE53FC">
<p>55. <strong>Unifying single-cell annotations based on the Cell Ontology</strong> <br />
Sheng Wang, Angela Oliveira Pisco, Aaron McGeever, Maria Brbic, Marinka Zitnik, Spyros Darmanis, Jure Leskovec, Jim Karkanias, Russ B. Altman<br />
<em>Cold Spring Harbor Laboratory</em> (2020-02-04) <a href="https://doi.org/gg9m2h">https://doi.org/gg9m2h</a> <br />
DOI: <a href="https://doi.org/10.1101/810234">10.1101/810234</a></p>
</div>
<div id="ref-l6iqnp9w">
<p>56. <strong>The naming of neurons: applications of taxonomic theory to the study of cellular populations.</strong> <br />
Tyner CF<br />
<em>Brain, Behavior and Evolution</em> (1975-01-01) <a href="https://www.wikidata.org/wiki/Q34065481">https://www.wikidata.org/wiki/Q34065481</a> <br />
DOI: <a href="https://doi.org/10.1159/000124141">10.1159/000124141</a></p>
</div>
<div id="ref-klRLs2Ss">
<p>57. <strong>Neuronal cell-type classification: challenges, opportunities and the path forward</strong> <br />
Hongkui Zeng, Joshua R. Sanes<br />
<em>Nature Reviews Neuroscience</em> (2017-08-03) <a href="https://doi.org/gbq3m8">https://doi.org/gbq3m8</a> <br />
DOI: <a href="https://doi.org/10.1038/nrn.2017.85">10.1038/nrn.2017.85</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28775344">28775344</a></p>
</div>
</div>
<!-- default theme -->

<style>
    /* import google fonts */
    @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
    @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

    /* -------------------------------------------------- */
    /* global */
    /* -------------------------------------------------- */

    /* all elements */
    * {
        /* force sans-serif font unless specified otherwise */
        font-family: "Open Sans", "Helvetica", sans-serif;

        /* prevent text inflation on some mobile browsers */
        -webkit-text-size-adjust: none !important;
        -moz-text-size-adjust: none !important;
        -o-text-size-adjust: none !important;
        text-size-adjust: none !important;
    }

    @media only screen {
        /* "page" element */
        body {
            position: relative;
            box-sizing: border-box;
            font-size: 12pt;
            line-height: 1.5;
            max-width: 8.5in;
            margin: 20px auto;
            padding: 40px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* "page" element */
        body {
            padding: 20px;
            margin: 0;
            border-radius: 0;
            border: none;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
            background: none;
        }
    }

    /* -------------------------------------------------- */
    /* headings */
    /* -------------------------------------------------- */

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        margin: 20px 0;
        padding: 0;
        font-weight: bold;
    }

    /* biggest heading */
    h1 {
        margin: 40px 0;
        text-align: center;
    }

    /* second biggest heading */
    h2 {
        margin-top: 30px;
        padding-bottom: 5px;
        border-bottom: solid 1px #bdbdbd;
    }

    /* heading font sizes */
    h1 {
        font-size: 2em;
    }
    h2 {
        font-size: 1.5em;
    }
    h3{
        font-size: 1.35em;
    }
    h4 {
        font-size: 1.25em;
    }
    h5 {
        font-size: 1.15em;
    }
    h6 {
        font-size: 1em;
    }

    /* -------------------------------------------------- */
    /* manuscript header */
    /* -------------------------------------------------- */

    /* manuscript title */
    header > h1 {
        margin: 0;
    }

    /* manuscript title caption text (ie "automatically generated on") */
    header + p {
        text-align: center;
        margin-top: 10px;
    }

    /* -------------------------------------------------- */
    /* text elements */
    /* -------------------------------------------------- */

    /* links */
    a {
        color: #2196f3;
        overflow-wrap: break-word;
    }

    /* normal links (not empty, not button link, not syntax highlighting link) */
    a:not(:empty):not(.button):not(.sourceLine) {
        padding-left: 1px;
        padding-right: 1px;
    }

    /* superscripts and subscripts */
    sub,
    sup {
        /* prevent from affecting line height */
        line-height: 0;
    }

    /* unordered and ordered lists*/
    ul,
    ol {
        padding-left: 20px;
    }

    /* class for styling text semibold */
    .semibold {
        font-weight: 600;
    }

    /* class for styling elements horizontally left aligned */
    .left {
        display: block;
        text-align: left;
        margin-left: auto;
        margin-right: 0;
        justify-content: left;
    }

    /* class for styling elements horizontally centered */
    .center {
        display: block;
        text-align: center;
        margin-left: auto;
        margin-right: auto;
        justify-content: center;
    }

    /* class for styling elements horizontally right aligned */
    .right {
        display: block;
        text-align: right;
        margin-left: 0;
        margin-right: auto;
        justify-content: right;
    }

    /* -------------------------------------------------- */
    /* section elements */
    /* -------------------------------------------------- */

    /* horizontal divider line */
    hr {
        border: none;
        height: 1px;
        background: #bdbdbd;
    }

    /* paragraphs, horizontal dividers, figures, tables, code */
    p,
    hr,
    figure,
    table,
    pre {
        /* treat all as "paragraphs", with consistent vertical margins */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* figures */
    /* -------------------------------------------------- */

    /* figure */
    figure {
        max-width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure caption */
    figcaption {
        padding: 0;
        padding-top: 10px;
    }

    /* figure image element */
    figure > img,
    figure > svg {
        max-width: 100%;
        display: block;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure auto-number */
    img + figcaption > span:first-of-type,
    svg + figcaption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* tables */
    /* -------------------------------------------------- */

    /* table */
    table {
        border-collapse: collapse;
        border-spacing: 0;
        width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* table cells */
    th,
    td {
        border: solid 1px #bdbdbd;
        padding: 10px;
        /* squash table if too wide for page by forcing line breaks */
        overflow-wrap: break-word;
        word-break: break-word;
    }

    /* header row and even rows */
    th,
    tr:nth-child(2n) {
        background-color: #fafafa;
    }

    /* odd rows */
    tr:nth-child(2n + 1) {
        background-color: #ffffff;
    }

    /* table caption */
    caption {
        text-align: left;
        padding: 0;
        padding-bottom: 10px;
    }

    /* table auto-number */
    table > caption > span:first-of-type,
    div.table_wrapper > table > caption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* code */
    /* -------------------------------------------------- */

    /* multi-line code block */
    pre {
        padding: 10px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
        break-inside: avoid;
        text-align: left;
    }

    /* inline code, ie code within normal text */
    :not(pre) > code {
        padding: 0 4px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
    }

    /* code text */
    /* apply all children, to reach syntax highlighting sub-elements */
    code,
    code * {
        /* force monospace font */
        font-family: "Source Code Pro", "Courier New", monospace;
    }

    /* -------------------------------------------------- */
    /* quotes */
    /* -------------------------------------------------- */

    /* quoted text */
    blockquote {
        margin: 0;
        padding: 0;
        border-left: 4px solid #bdbdbd;
        padding-left: 16px;
        break-inside: avoid;
    }

    /* -------------------------------------------------- */
    /* banners */
    /* -------------------------------------------------- */

    /* info banners */
    .banner {
        box-sizing: border-box;
        display: block;
        position: relative;
        width: 100%;
        margin-top: 20px;
        margin-bottom: 20px;
        padding: 20px;
        text-align: center;
    }

    /* paragraph in banner */
    .banner > p {
        margin: 0;
    }

    /* -------------------------------------------------- */
    /* highlight colors */
    /* -------------------------------------------------- */

    .white {
        background: #ffffff;
    }
    .lightgrey {
        background: #eeeeee;
    }
    .grey {
        background: #757575;
    }
    .darkgrey {
        background: #424242;
    }
    .black {
        background: #000000;
    }
    .lightred {
        background: #ffcdd2;
    }
    .lightyellow {
        background: #ffecb3;
    }
    .lightgreen {
        background: #dcedc8;
    }
    .lightblue {
        background: #e3f2fd;
    }
    .lightpurple {
        background: #f3e5f5;
    }
    .red {
        background: #f44336;
    }
    .orange {
        background: #ff9800;
    }
    .yellow {
        background: #ffeb3b;
    }
    .green {
        background: #4caf50;
    }
    .blue {
        background: #2196f3;
    }
    .purple {
        background: #9c27b0;
    }
    .white,
    .lightgrey,
    .lightred,
    .lightyellow,
    .lightgreen,
    .lightblue,
    .lightpurple,
    .orange,
    .yellow,
    .white a,
    .lightgrey a,
    .lightred a,
    .lightyellow a,
    .lightgreen a,
    .lightblue a,
    .lightpurple a,
    .orange a,
    .yellow a {
        color: #000000;
    }
    .grey,
    .darkgrey,
    .black,
    .red,
    .green,
    .blue,
    .purple,
    .grey a,
    .darkgrey a,
    .black a,
    .red a,
    .green a,
    .blue a,
    .purple a {
        color: #ffffff;
    }

    /* -------------------------------------------------- */
    /* buttons */
    /* -------------------------------------------------- */

    /* class for styling links like buttons */
    .button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        margin: 5px;
        padding: 10px 20px;
        font-size: 0.75em;
        font-weight: 600;
        text-transform: uppercase;
        text-decoration: none;
        letter-spacing: 1px;
        background: none;
        color: #2196f3;
        border: solid 1px #bdbdbd;
        border-radius: 5px;
    }

    /* buttons when hovered */
    .button:hover:not([disabled]),
    .icon_button:hover:not([disabled]) {
        cursor: pointer;
        background: #f5f5f5;
    }

    /* buttons when disabled */
    .button[disabled],
    .icon_button[disabled] {
        opacity: 0.35;
        pointer-events: none;
    }

    /* class for styling buttons containg only single icon */
    .icon_button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        text-decoration: none;
        margin: 0;
        padding: 0;
        background: none;
        border-radius: 5px;
        border: none;
        width: 20px;
        height: 20px;
        min-width: 20px;
        min-height: 20px;
    }

    /* icon button inner svg image */
    .icon_button > svg {
        height: 16px;
    }

    /* -------------------------------------------------- */
    /* icons */
    /* -------------------------------------------------- */

    /* class for styling icons inline with text */
    .inline_icon {
        height: 1em;
        position: relative;
        top: 0.125em;
    }

    /* -------------------------------------------------- */
    /* print control */
    /* -------------------------------------------------- */

    @media print {
        @page {
            /* suggested printing margin */
            margin: 0.5in;
        }

        /* document and "page" elements */
        html, body {
            margin: 0;
            padding: 0;
            width: 100%;
            height: 100%;
        }

        /* "page" element */
        body {
            font-size: 11pt !important;
            line-height: 1.35;
        }

        /* all headings */
        h1,
        h2,
        h3,
        h4,
        h5,
        h6 {
            margin: 15px 0;
        }

        /* figures and tables */
        figure, table {
            font-size: 0.85em;
        }

        /* table cells */
        th,
        td {
            padding: 5px;
        }

        /* shrink font awesome icons */
        i.fas,
        i.fab,
        i.far,
        i.fal {
            transform: scale(0.85);
        }

        /* decrease banner margins */
        .banner {
            margin-top: 15px;
            margin-bottom: 15px;
            padding: 15px;
        }

        /* class for centering an element vertically on its own page */
        .page_center {
            margin: auto;
            width: 100%;
            height: 100%;
            display: flex;
            align-items: center;
            vertical-align: middle;
            break-before: page;
            break-after: page;
        }

        /* always insert a page break before the element */
        .page_break_before {
            break-before: page;
        }

        /* always insert a page break after the element */
        .page_break_after {
            break-after: page;
        }

        /* avoid page break before the element */
        .page_break_before_avoid {
            break-before: avoid;
        }

        /* avoid page break after the element */
        .page_break_after_avoid {
            break-after: avoid;
        }

        /* avoid page break inside the element */
        .page_break_inside_avoid {
            break-inside: avoid;
        }
    }

    /* -------------------------------------------------- */
    /* override pandoc css quirks */
    /* -------------------------------------------------- */

    .sourceCode {
        /* prevent unsightly overflow in wide code blocks */
        overflow: auto !important;
    }

    div.sourceCode {
        /* prevent background fill on top-most code block  container */
        background: none !important;
    }

    .sourceCode * {
        /* force consistent line spacing */
        line-height: 1.5 !important;
    }

    div.sourceCode {
        /* style code block margins same as <pre> element */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* tablenos */
    /* -------------------------------------------------- */

    /* tablenos wrapper */
    .tablenos {
        /* show scrollbar on tables if necessary to prevent overflow */
        width: 100%;
        margin: 20px 0;
    }

    .tablenos > table {
        /* move margins from table to table_wrapper to allow margin collapsing */
        margin: 0;
    }

    @media only screen {
        /* tablenos wrapper */
        .tablenos {
            /* show scrollbar on tables if necessary to prevent overflow */
            overflow-x: auto !important;
        }

        .tablenos th,
        .tablenos td {
            overflow-wrap: unset !important;
            word-break: unset !important;
        }

        /* table in wrapper */
        .tablenos table,
        .tablenos table * {
            /* don't break table words */
            overflow-wrap: normal !important;
        }
    }

    /* -------------------------------------------------- */
    /* mathjax */
    /* -------------------------------------------------- */

    /* mathjax containers */
    .math.display > span:not(.MathJax_Preview) {
        /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
        display: flex !important;
        overflow-x: auto !important;
        overflow-y: hidden !important;
        justify-content: center;
        align-items: center;
        margin: 0 !important;
    }

    /* right click menu */
    .MathJax_Menu {
        border-radius: 5px !important;
        border: solid 1px #bdbdbd !important;
        box-shadow: none !important;
    }

    /* equation auto-number */
    span[id^="eq:"] > span.math.display + span {
        font-weight: 600;
    }

    /* equation */
    span[id^="eq:"] > span.math.display > span {
        /* nudge to make room for equation auto-number and anchor */
        margin-right: 60px !important;
    }

    /* -------------------------------------------------- */
    /* anchors plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anchor button */
        .anchor {
            opacity: 0;
            margin-left: 5px;
        }

        /* anchor buttons within <h2>'s */
        h2 .anchor {
            margin-left: 10px;
        }

        /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
        *:hover > .anchor,
        .anchor:hover,
        .anchor:focus {
            opacity: 1;
        }

        /* anchor button when hovered */
        .anchor:hover {
            cursor: pointer;
        }
    }

    /* always show anchor button on devices with no mouse/hover ability */
    @media (hover: none) {
        .anchor {
            opacity: 1;
        }
    }

    /* always hide anchor button on print */
    @media only print {
        .anchor {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* accordion plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* accordion arrow button */
        .accordion_arrow {
            margin-right: 10px;
        }

        /* arrow icon when <h2> data-collapsed attribute true */
        h2[data-collapsed="true"] > .accordion_arrow > svg {
            transform: rotate(-90deg);
        }

        /* all elements (except <h2>'s) when data-collapsed attribute true */
        *:not(h2)[data-collapsed="true"] {
            display: none;
        }

        /* accordion arrow button when hovered and <h2>'s when hovered */
        .accordion_arrow:hover,
        h2[data-collapsed="true"]:hover,
        h2[data-collapsed="false"]:hover {
            cursor: pointer;
        }
    }

    /* always hide accordion arrow button on print */
    @media only print {
        .accordion_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* tooltips plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* tooltip container */
        #tooltip {
            position: absolute;
            width: 50%;
            min-width: 240px;
            max-width: 75%;
            z-index: 1;
        }

        /* tooltip content */
        #tooltip_content {
            margin-bottom: 5px;
            padding: 20px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
            overflow-wrap: break-word;
        }

        /* tooltip copy of paragraphs and figures */
        #tooltip_content > p,
        #tooltip_content > figure {
            margin: 0;
            max-height: 320px;
            overflow-y: auto;
        }

        /* tooltip copy of <img> */
        #tooltip_content > figure > img,
        #tooltip_content > figure > svg {
            max-height: 260px;
        }

        /* navigation bar */
        #tooltip_nav_bar {
            margin-top: 10px;
            text-align: center;
        }

        /* navigation bar previous/next buton */
        #tooltip_nav_bar > .icon_button {
            position: relative;
            top: 3px;
        }

        /* navigation bar previous button */
        #tooltip_nav_bar > .icon_button:first-of-type {
            margin-right: 5px;
        }

        /* navigation bar next button */
        #tooltip_nav_bar > .icon_button:last-of-type {
            margin-left: 5px;
        }
    }

    /* always hide tooltip on print */
    @media only print {
        #tooltip {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* jump to first plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* jump button */
        .jump_arrow {
            position: relative;
            top: 0.125em;
            margin-right: 5px;
        }
    }

    /* always hide jump button on print */
    @media only print {
        .jump_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* link highlight plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anything with data-highlighted attribute true */
        [data-highlighted="true"] {
            background: #ffeb3b;
        }

        /* anything with data-selected attribute true */
        [data-selected="true"] {
            background: #ff8a65 !important;
        }

        /* animation definition for glow */
        @keyframes highlight_glow {
            0% {
                background: none;
            }
            10% {
                background: #bbdefb;
            }
            100% {
                background: none;
            }
        }

        /* anything with data-glow attribute true */
        [data-glow="true"] {
            animation: highlight_glow 2s;
        }
    }

    /* -------------------------------------------------- */
    /* table of contents plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* toc panel */
        #toc_panel {
            box-sizing: border-box;
            position: fixed;
            top: 0;
            left: 0;
            background: #ffffff;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            z-index: 2;
        }

        /* toc panel when closed */
        #toc_panel[data-open="false"] {
            min-width: 60px;
            width: 60px;
            height: 60px;
            border-right: solid 1px #bdbdbd;
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc panel when open */
        #toc_panel[data-open="true"] {
            min-width: 260px;
            max-width: 480px;
            /* keep panel edge consistent distance away from "page" edge */
            width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
            bottom: 0;
            border-right: solid 1px #bdbdbd;
        }

        /* toc panel header */
        #toc_header {
            box-sizing: border-box;
            display: flex;
            flex-direction: row;
            align-items: center;
            height: 60px;
            margin: 0;
            padding: 20px;
        }

        /* toc panel header when hovered */
        #toc_header:hover {
            cursor: pointer;
        }

        /* toc panel header when panel open */
        #toc_panel[data-open="true"] > #toc_header {
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc open/close header button */
        #toc_button {
            margin-right: 20px;
        }

        /* hide toc list and header text when closed */
        #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
        #toc_panel[data-open="false"] > #toc_list {
            display: none;
        }

        /* toc list of entries */
        #toc_list {
            box-sizing: border-box;
            width: 100%;
            padding: 20px;
            position: absolute;
            top: calc(60px + 1px);
            bottom: 0;
            overflow: auto;
        }

        /* toc entry, link to section in document */
        .toc_link {
            display: block;
            padding: 5px;
            position: relative;
            font-weight: 600;
            text-decoration: none;
        }

        /* toc entry when hovered or when "viewed" */
        .toc_link:hover,
        .toc_link[data-viewing="true"] {
            background: #f5f5f5;
        }

        /* toc entry, level 1 indentation */
        .toc_link[data-level="1"] {
            margin-left: 0;
        }

        /* toc entry, level 2 indentation */
        .toc_link[data-level="2"] {
            margin-left: 20px;
        }

        /* toc entry, level 3 indentation */
        .toc_link[data-level="3"] {
            margin-left: 40px;
        }

        /* toc entry, level 4 indentation */
        .toc_link[data-level="4"] {
            margin-left: 60px;
        }

        /* toc entry bullets */
        #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
            position: absolute;
            left: -15px;
            top: -1px;
            font-size: 1.5em;
        }

        /* toc entry, level 2 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
            content: "\2022";
        }

        /* toc entry, level 3 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
            content: "\25AB";
        }

        /* toc entry, level 4 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
            content: "-";
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* push <body> ("page") element down to make room for toc icon */
        .toc_body_nudge {
            padding-top: 60px;
        }

        /* toc icon when panel closed and not hovered */
        #toc_panel[data-open="false"]:not(:hover) {
            background: rgba(255, 255, 255, 0.75);
        }
    }

    /* always hide toc panel on print */
    @media only print {
        #toc_panel {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* lightbox plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* regular <img> in document when hovered */
        img.lightbox_document_img:hover {
            cursor: pointer;
        }

        .body_no_scroll {
            overflow: hidden !important;
        }

        /* screen overlay */
        #lightbox_overlay {
            display: flex;
            flex-direction: column;
            position: fixed;
            left: 0;
            top: 0;
            right: 0;
            bottom: 0;
            background: rgba(0, 0, 0, 0.75);
            z-index: 3;
        }

        /* middle area containing lightbox image */
        #lightbox_image_container {
            flex-grow: 1;
            display: flex;
            justify-content: center;
            align-items: center;
            overflow: hidden;
            position: relative;
            padding: 20px;
        }

        /* bottom area containing caption */
        #lightbox_bottom_container {
            display: flex;
            justify-content: center;
            align-items: center;
            height: 100px;
            min-height: 100px;
            max-height: 100px;
            background: rgba(0, 0, 0, 0.5);
        }

        /* image number info text box */
        #lightbox_number_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            left: 2px;
            top: 0;
            z-index: 4;
        }

        /* zoom info text box */
        #lightbox_zoom_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            right: 2px;
            top: 0;
            z-index: 4;
        }

        /* copy of image caption */
        #lightbox_caption {
            box-sizing: border-box;
            display: inline-block;
            width: 100%;
            max-height: 100%;
            padding: 10px 0;
            text-align: center;
            overflow-y: auto;
            color: #ffffff;
        }

        /* navigation previous/next button */
        .lightbox_button {
            width: 100px;
            height: 100%;
            min-width: 100px;
            min-height: 100%;
            color: #ffffff;
        }

        /* navigation previous/next button when hovered */
        .lightbox_button:hover {
            background: none !important;
        }

        /* navigation button icon */
        .lightbox_button > svg {
            height: 25px;
        }

        /* figure auto-number */
        #lightbox_caption > span:first-of-type {
            font-weight: bold;
            margin-right: 5px;
        }

        /* lightbox image when hovered */
        #lightbox_img:hover {
            cursor: grab;
        }

        /* lightbox image when grabbed */
        #lightbox_img:active {
            cursor: grabbing;
        }
    }

    /* when on screen < 480px wide */
    @media only screen and (max-width: 480px) {
        /* make navigation buttons skinnier on small screens to make more room for caption text */
        .lightbox_button {
            width: 50px;
            min-width: 50px;
        }
    }

    /* always hide lightbox on print */
    @media only print {
        #lightbox_overlay {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* hypothesis (annotations) plugin */
    /* -------------------------------------------------- */

    /* hypothesis activation button */
    #hypothesis_button {
        box-sizing: border-box;
        position: fixed;
        top: 0;
        right: 0;
        width: 60px;
        height: 60px;
        background: #ffffff;
        border-radius: 0;
        border-left: solid 1px #bdbdbd;
        border-bottom: solid 1px #bdbdbd;
        box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
        z-index: 2;
    }

    /* hypothesis button svg */
    #hypothesis_button > svg {
        position: relative;
        top: -4px;
    }

    /* hypothesis annotation count */
    #hypothesis_count {
        position: absolute;
        left: 0;
        right: 0;
        bottom: 5px;
    }

    /* side panel */
    .annotator-frame {
        width: 280px !important;
    }

    /* match highlight color to rest of theme */
    .annotator-highlights-always-on .annotator-hl {
        background-color: #ffeb3b !important;
    }

    /* match focused color to rest of theme */
    .annotator-hl.annotator-hl-focused {
        background-color: #ff8a65 !important;
    }

    /* match bucket bar color to rest of theme */
    .annotator-bucket-bar {
        background: #f5f5f5 !important;
    }

    /* always hide button, toolbar, and tooltip on print */
    @media only print {
        #hypothesis_button {
            display: none;
        }

        .annotator-frame {
            display: none !important;
        }

        hypothesis-adder {
            display: none !important;
        }
    }
</style>
<!-- anchors plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds an anchor next to each of a certain type
        // of element that provides a human-readable url to that specific
        // item/position in the document (eg "manuscript.html#abstract"). It
        // also makes it such that scrolling out of view of a target removes
        // its identifier from the url.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'anchors';

        // default plugin options
        const options = {
            // which types of elements to add anchors next to, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3, [id^="fig:"], [id^="tbl:"], [id^="eq:"]',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // add anchor to each element of specified types
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements)
                addAnchor(element);

            // attach scroll listener to window
            window.addEventListener('scroll', onScroll);
        }

        // when window is scrolled
        function onScroll() {
            // if url has hash and user has scrolled out of view of hash
            // target, remove hash from url
            const tolerance = 100;
            const target = getHashTarget();
            if (target) {
                if (
                    target.getBoundingClientRect().top >
                        window.innerHeight + tolerance ||
                    target.getBoundingClientRect().bottom < 0 - tolerance
                )
                    history.pushState(null, null, ' ');
            }
        }

        // add anchor to element
        function addAnchor(element) {
            let addTo; // element to add anchor button to

            // if figure or table, modify withId and addTo to get expected
            // elements
            if (element.id.indexOf('fig:') === 0) {
                addTo = element.querySelector('figcaption');
            } else if (element.id.indexOf('tbl:') === 0) {
                addTo = element.querySelector('caption');
            } else if (element.id.indexOf('eq:') === 0) {
                addTo = element.querySelector('.eqnos-number');
            }

            addTo = addTo || element;
            const id = element.id || null;

            // do not add anchor if element doesn't have assigned id.
            // id is generated by pandoc and is assumed to be unique and
            // human-readable
            if (!id)
                return;

            // create anchor button
            const anchor = document.createElement('a');
            anchor.innerHTML = document.querySelector('.icon_link').innerHTML;
            anchor.title = 'Link to this part of the document';
            anchor.classList.add('icon_button', 'anchor');
            anchor.dataset.ignore = 'true';
            anchor.href = '#' + id;
            addTo.appendChild(anchor);
        }

        // get element that is target of link or url hash
        function getHashTarget() {
            const hash = window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');
            if (id.indexOf('tbl:') === 0)
                target = target.querySelector('table');

            return target;
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- link icon -->

<template class="icon_link">
    <!-- modified from: https://fontawesome.com/icons/link -->
    <svg width="16" height="16" viewBox="0 0 512 512">
        <path
            fill="currentColor"
            d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
        ></path>
    </svg>
</template>
<!-- accordion plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows sections of content under <h2> headings
        // to be collapsible.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'accordion';

        // default plugin options
        const options = {
            // whether to always start expanded ('false'), always start
            // collapsed ('true'), or start collapsed when screen small ('auto')
            startCollapsed: 'auto',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <h2> heading
            const headings = document.querySelectorAll('h2');
            for (const heading of headings) {
                addArrow(heading);

                // start expanded/collapsed based on option
                if (
                    options.startCollapsed === 'true' ||
                    (options.startCollapsed === 'auto' && isSmallScreen())
                )
                    collapseHeading(heading);
                else
                    expandHeading(heading);
            }

            // attach hash change listener to window
            window.addEventListener('hashchange', onHashChange);
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                goToElement(target);
        }

        // add arrow to heading
        function addArrow(heading) {
            // add arrow button
            const arrow = document.createElement('button');
            arrow.innerHTML = document.querySelector(
                '.icon_angle_down'
            ).innerHTML;
            arrow.classList.add('icon_button', 'accordion_arrow');
            heading.insertBefore(arrow, heading.firstChild);

            // attach click listener to heading and button
            heading.addEventListener('click', onHeadingClick);
            arrow.addEventListener('click', onArrowClick);
        }

        // determine if on mobile-like device with small screen
        function isSmallScreen() {
            return Math.min(window.innerWidth, window.innerHeight) < 480;
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get element that is target of hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');
            if (id.indexOf('tbl:') === 0)
                target = target.querySelector('table');

            return target;
        }

        // when <h2> heading is clicked
        function onHeadingClick(event) {
            // only collapse if <h2> itself is target of click (eg, user did
            // not click on anchor within <h2>)
            if (event.target === this)
                toggleCollapse(this);
        }

        // when arrow button is clicked
        function onArrowClick() {
            toggleCollapse(this.parentNode);
        }

        // collapse section if expanded, expand if collapsed
        function toggleCollapse(heading) {
            if (heading.dataset.collapsed === 'false')
                collapseHeading(heading);
            else
                expandHeading(heading);
        }

        // elements to exclude from collapse, such as table of contents panel,
        // hypothesis panel, etc
        const exclude = '#toc_panel, div.annotator-frame, #lightbox_overlay';

        // collapse section
        function collapseHeading(heading) {
            heading.setAttribute('data-collapsed', 'true');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'true');
        }

        // expand section
        function expandHeading(heading) {
            heading.setAttribute('data-collapsed', 'false');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'false');
        }

        // get list of elements between this <h2> and next <h2> or <h1>
        // ("children" of the <h2> section)
        function getChildren(heading) {
            return nextUntil(heading, 'h2, h1', exclude);
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get list of elements after a start element up to element matching
        // query
        function nextUntil(element, query, exclude) {
            const elements = [];
            while (element = element.nextElementSibling, element) {
                if (element.matches(query))
                    break;
                if (!element.matches(exclude))
                    elements.push(element);
            }
            return elements;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
    <!-- modified from: https://fontawesome.com/icons/angle-down -->
    <svg width="16" height="16" viewBox="0 0 448 512">
        <path
            fill="currentColor"
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        ></path>
    </svg>
</template>
<!-- tooltips plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when the user hovers or
        // focuses a link to a citation or figure, a tooltip appears with a
        // preview of the reference content, along with arrows to navigate
        // between instances of the same reference in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tooltips';

        // default plugin options
        const options = {
            // whether user must click off to close tooltip instead of just
            // un-hovering
            clickClose: 'false',
            // delay (in ms) between opening and closing tooltip
            delay: '100',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach hover and focus listeners to link
                link.addEventListener('mouseover', onLinkHover);
                link.addEventListener('mouseleave', onLinkUnhover);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('touchend', onLinkTouch);
            }

            // attach mouse, key, and resize listeners to window
            window.addEventListener('mousedown', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('keyup', onKeyUp);
            window.addEventListener('resize', onResize);
        }

        // when link is hovered
        function onLinkHover() {
            // function to open tooltip
            const delayOpenTooltip = function() {
                openTooltip(this);
            }.bind(this);

            // run open function after delay
            this.openTooltipTimer = window.setTimeout(
                delayOpenTooltip,
                options.delay
            );
        }

        // when mouse leaves link
        function onLinkUnhover() {
            // cancel opening tooltip
            window.clearTimeout(this.openTooltipTimer);

            // don't close on unhover if option specifies
            if (options.clickClose === 'true')
                return;

            // function to close tooltip
            const delayCloseTooltip = function() {
                // if tooltip open and if mouse isn't over tooltip, close
                const tooltip = document.getElementById('tooltip');
                if (tooltip && !tooltip.matches(':hover'))
                    closeTooltip();
            };

            // run close function after delay
            this.closeTooltipTimer = window.setTimeout(
                delayCloseTooltip,
                options.delay
            );
        }

        // when link is focused (tabbed to)
        function onLinkFocus(event) {
            openTooltip(this);
        }

        // when link is touched on touch screen
        function onLinkTouch(event) {
            // attempt to force hover state on first tap always, and trigger
            // regular link click (and navigation) on second tap
            if (event.target === document.activeElement)
                event.target.click();
            else {
                document.activeElement.blur();
                event.target.focus();
            }
            if (event.cancelable)
                event.preventDefault();
            event.stopPropagation();
            return false;
        }

        // when mouse is clicked anywhere in window
        function onClick(event) {
            closeTooltip();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'tooltip_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'tooltip_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeTooltip();
                    break;
            }
        }

        // when window is resized or zoomed
        function onResize() {
            closeTooltip();
        }

        // get all links of types we wish to handle
        function getLinks() {
            const queries = [];
            // exclude buttons, anchor links, toc links, etc
            const exclude =
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            queries.push('a[href^="#ref-"]' + exclude); // citation links
            queries.push('a[href^="#fig:"]' + exclude); // figure links
            const query = queries.join(', ');
            return document.querySelectorAll(query);
        }

        // get links with same target, get index of link in set, get total
        // same links
        function getSameLinks(link) {
            const sameLinks = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    sameLinks.push(otherLink);
            }

            return {
                elements: sameLinks,
                index: sameLinks.indexOf(link),
                total: sameLinks.length
            };
        }

        // open tooltip
        function openTooltip(link) {
            // delete tooltip if it exists, start fresh
            closeTooltip();

            // make tooltip element
            const tooltip = makeTooltip(link);

            // if source couldn't be found and tooltip not made, exit
            if (!tooltip)
                return;

            // make navbar elements
            const navBar = makeNavBar(link);
            if (navBar)
                tooltip.firstElementChild.appendChild(navBar);

            // attach tooltip to page
            document.body.appendChild(tooltip);

            // position tooltip
            const position = function() {
                positionTooltip(link);
            };
            position();

            // if tooltip contains images, position again after they've loaded
            const imgs = tooltip.querySelectorAll('img');
            for (const img of imgs)
                img.addEventListener('load', position);
        }

        // close (delete) tooltip
        function closeTooltip() {
            const tooltip = document.getElementById('tooltip');
            if (tooltip)
                tooltip.remove();
        }

        // make tooltip
        function makeTooltip(link) {
            // get target element that link points to
            const source = getSource(link);

            // if source can't be found, exit
            if (!source)
                return;

            // create new tooltip
            const tooltip = document.createElement('div');
            tooltip.id = 'tooltip';
            const tooltipContent = document.createElement('div');
            tooltipContent.id = 'tooltip_content';
            tooltip.appendChild(tooltipContent);

            // make copy of source node and put in tooltip
            const sourceCopy = makeCopy(source);
            tooltipContent.appendChild(sourceCopy);

            // attach mouse event listeners
            tooltip.addEventListener('click', onTooltipClick);
            tooltip.addEventListener('mousedown', onTooltipClick);
            tooltip.addEventListener('touchstart', onTooltipClick);
            tooltip.addEventListener('mouseleave', onTooltipUnhover);

            // (for interaction with lightbox plugin)
            // transfer click on tooltip copied img to original img
            const sourceImg = source.querySelector('img');
            const sourceCopyImg = sourceCopy.querySelector('img');
            if (sourceImg && sourceCopyImg) {
                const clickImg = function() {
                    sourceImg.click();
                    closeTooltip();
                };
                sourceCopyImg.addEventListener('click', clickImg);
            }

            return tooltip;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // when tooltip is clicked
        function onTooltipClick(event) {
            // when user clicks on tooltip, stop click from transferring
            // outside of tooltip (eg, click off to close tooltip, or eg click
            // off to unhighlight same refs)
            event.stopPropagation();
        }

        // when tooltip is unhovered
        function onTooltipUnhover(event) {
            if (options.clickClose === 'true')
                return;

            // make sure new mouse/touch/focus no longer over tooltip or any
            // element within it
            const tooltip = document.getElementById('tooltip');
            if (!tooltip)
                return;
            if (this.contains(event.relatedTarget))
                return;

            closeTooltip();
        }

        // make nav bar to go betwen prev/next instances of same reference
        function makeNavBar(link) {
            // find other links to the same source
            const sameLinks = getSameLinks(link);

            // don't show nav bar when singular reference
            if (sameLinks.total <= 1)
                return;

            // find prev/next links with same target
            const prevLink = getPrevLink(link, sameLinks);
            const nextLink = getNextLink(link, sameLinks);

            // create nav bar
            const navBar = document.createElement('div');
            navBar.id = 'tooltip_nav_bar';
            const text = sameLinks.index + 1 + ' of ' + sameLinks.total;

            // create nav bar prev/next buttons
            const prevButton = document.createElement('button');
            const nextButton = document.createElement('button');
            prevButton.id = 'tooltip_prev_button';
            nextButton.id = 'tooltip_next_button';
            prevButton.title =
                'Jump to the previous occurence of this item in the document [←]';
            nextButton.title =
                'Jump to the next occurence of this item in the document [→]';
            prevButton.classList.add('icon_button');
            nextButton.classList.add('icon_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;
            navBar.appendChild(prevButton);
            navBar.appendChild(document.createTextNode(text));
            navBar.appendChild(nextButton);

            // attach click listeners to buttons
            prevButton.addEventListener('click', function() {
                onPrevNextClick(link, prevLink);
            });
            nextButton.addEventListener('click', function() {
                onPrevNextClick(link, nextLink);
            });

            return navBar;
        }

        // get previous link with same target
        function getPrevLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if < 1
            let index;
            if (sameLinks.index - 1 >= 0)
                index = sameLinks.index - 1;
            else
                index = sameLinks.total - 1;
            return sameLinks.elements[index];
        }

        // get next link with same target
        function getNextLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if > total
            let index;
            if (sameLinks.index + 1 <= sameLinks.total - 1)
                index = sameLinks.index + 1;
            else
                index = 0;
            return sameLinks.elements[index];
        }

        // get element that is target of link or url hash
        function getSource(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if ref or figure, modify target to get expected element
            if (id.indexOf('ref-') === 0)
                target = target.querySelector('p');
            else if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');

            return target;
        }

        // when prev/next arrow button is clicked
        function onPrevNextClick(link, prevNextLink) {
            if (link && prevNextLink)
                goToElement(prevNextLink, window.innerHeight * 0.5);
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // determine position to place tooltip based on link position in
        // viewport and tooltip size
        function positionTooltip(link, left, top) {
            const tooltipElement = document.getElementById('tooltip');
            if (!tooltipElement)
                return;

            // get convenient vars for position/dimensions of
            // link/tooltip/page/view
            link = getRectInPage(link);
            const tooltip = getRectInPage(tooltipElement);
            const view = getRectInPage();

            // horizontal positioning
            if (left)
                // use explicit value
                left = left;
            else if (link.left + tooltip.width < view.right)
                // fit tooltip to right of link
                left = link.left;
            else if (link.right - tooltip.width > view.left)
                // fit tooltip to left of link
                left = link.right - tooltip.width;
            // center tooltip in view
            else
                left = (view.right - view.left) / 2 - tooltip.width / 2;

            // vertical positioning
            if (top)
                // use explicit value
                top = top;
            else if (link.top - tooltip.height > view.top)
                // fit tooltip above link
                top = link.top - tooltip.height;
            else if (link.bottom + tooltip.height < view.bottom)
                // fit tooltip below link
                top = link.bottom;
            else {
                // center tooltip in view
                top = view.top + view.height / 2 - tooltip.height / 2;
                // nudge off of link to left/right if possible
                if (link.right + tooltip.width < view.right)
                    left = link.right;
                else if (link.left - tooltip.width > view.left)
                    left = link.left - tooltip.width;
            }

            tooltipElement.style.left = left + 'px';
            tooltipElement.style.top = top + 'px';
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get position of element relative to page
        function getRectInPage(element) {
            const rect = getRectInView(element);
            const body = getRectInView(document.body);

            const newRect = {};
            newRect.left = rect.left - body.left;
            newRect.top = rect.top - body.top;
            newRect.right = rect.right - body.left;
            newRect.bottom = rect.bottom - body.top;
            newRect.width = rect.width;
            newRect.height = rect.height;

            return newRect;
        }

        // (for interaction with accordion plugin)
        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // (for interaction with accordion plugin)
        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- jump to first plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds a button next to each reference entry,
        // figure, and table that jumps the page to the first occurrence of a
        // link to that item in the manuscript.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'jumpToFirst';

        // default plugin options
        const options = {
            // whether to add buttons next to reference entries
            references: 'true',
            // whether to add buttons next to figures
            figures: 'true',
            // whether to add buttons next to tables
            tables: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            if (options.references !== 'false')
                makeReferenceButtons();
            if (options.figures !== 'false')
                makeFigureButtons();
            if (options.tables !== 'false')
                makeTableButtons();
        }

        // when jump button clicked
        function onButtonClick() {
            const first = getFirstOccurrence(this.dataset.id);
            if (!first)
                return;

            // update url hash so navigating "back" in history will return
            // user to jump button
            window.location.hash = this.dataset.id;
            // scroll to link
            window.setTimeout(function() {
                goToElement(first, window.innerHeight * 0.5);
            }, 0);
        }

        // get first occurence of link to item in document
        function getFirstOccurrence(id) {
            let query = 'a';
            query += '[href="#' + id + '"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelector(query);
        }

        // add button next to each reference entry
        function makeReferenceButtons() {
            const references = document.querySelectorAll('div[id^="ref-"]');
            for (const reference of references) {
                // get reference id and element to add button to
                const id = reference.id;
                const container = reference.firstElementChild;
                const first = getFirstOccurrence(id);

                // if can't find link to reference, ignore
                if (!first)
                    continue;

                // make jump button
                let button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this reference in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.innerHTML = button.outerHTML + container.innerHTML;
                button = container.firstElementChild;
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeFigureButtons() {
            const figures = document.querySelectorAll('[id^="fig:"]');
            for (const figure of figures) {
                // get figure id and element to add button to
                const id = figure.id;
                const container = figure.querySelector('figcaption') || figure;
                const first = getFirstOccurrence(id);

                // if can't find link to figure, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this figure in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeTableButtons() {
            const tables = document.querySelectorAll('[id^="tbl:"]');
            for (const table of tables) {
                // get ref id and element to add button to
                const id = table.id;
                const container = table.querySelector('caption') || table;
                const first = getFirstOccurrence(id);

                // if can't find link to table, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this table in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
    <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
    <svg width="16" height="16" viewBox="0 0 320 512">
        <path
            fill="currentColor"
            d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
        ></path>
    </svg>
</template>
<!-- link highlight plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user hovers or
        // focuses a link, other links that have the same target will be
        // highlighted. It also makes it such that when clicking a link, the
        // target of the link (eg reference, figure, table) is briefly
        // highlighted.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'linkHighlight';

        // default plugin options
        const options = {
            // whether to also highlight links that go to external urls
            externalLinks: 'false',
            // whether user must click off to unhighlight instead of just
            // un-hovering
            clickUnhighlight: 'false',
            // whether to also highlight links that are unique
            highlightUnique: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach mouse and focus listeners to link
                link.addEventListener('mouseenter', onLinkFocus);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('mouseleave', onLinkUnhover);
            }

            // attach click and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('hashchange', onHashChange);

            // run hash change on window load in case user has navigated
            // directly to hash
            onHashChange();
        }

        // when link is focused (tabbed to) or hovered
        function onLinkFocus() {
            highlight(this);
        }

        // when link is unhovered
        function onLinkUnhover() {
            if (options.clickUnhighlight !== 'true')
                unhighlightAll();
        }

        // when the mouse is clicked anywhere in window
        function onClick(event) {
            unhighlightAll();
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                glowElement(target);
        }

        // get element that is target of link or url hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            return target;
        }

        // start glow sequence on an element
        function glowElement(element) {
            const startGlow = function() {
                onGlowEnd();
                element.dataset.glow = 'true';
                element.addEventListener('animationend', onGlowEnd);
            };
            const onGlowEnd = function() {
                element.removeAttribute('data-glow');
                element.removeEventListener('animationend', onGlowEnd);
            };
            startGlow();
        }

        // highlight link and all others with same target
        function highlight(link) {
            // force unhighlight all to start fresh
            unhighlightAll();

            // get links with same target
            if (!link)
                return;
            const sameLinks = getSameLinks(link);

            // if link unique and option is off, exit and don't highlight
            if (sameLinks.length <= 1 && options.highlightUnique !== 'true')
                return;

            // highlight all same links, and "select" (special highlight) this
            // one
            for (const sameLink of sameLinks) {
                if (sameLink === link)
                    sameLink.setAttribute('data-selected', 'true');
                else
                    sameLink.setAttribute('data-highlighted', 'true');
            }
        }

        // unhighlight all links
        function unhighlightAll() {
            const links = getLinks();
            for (const link of links) {
                link.setAttribute('data-selected', 'false');
                link.setAttribute('data-highlighted', 'false');
            }
        }

        // get links with same target
        function getSameLinks(link) {
            const results = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    results.push(otherLink);
            }
            return results;
        }

        // get all links of types we wish to handle
        function getLinks() {
            let query = 'a';
            if (options.externalLinks !== 'true')
                query += '[href^="#"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelectorAll(query);
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin provides a "table of contents" (toc) panel on
        // the side of the document that allows the user to conveniently
        // navigate between sections of the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tableOfContents';

        // default plugin options
        const options = {
            // which types of elements to add links for, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3',
            // whether toc starts open. use 'true' or 'false', or 'auto' to
            // use 'true' behavior when screen wide enough and 'false' when not
            startOpen: 'false',
            // whether toc closes when clicking on toc link. use 'true' or
            // 'false', or 'auto' to use 'false' behavior when screen wide
            // enough and 'true' when not
            clickClose: 'auto',
            // if list item is more than this many characters, text will be
            // truncated
            charLimit: '50',
            // whether or not to show bullets next to each toc item
            bullets: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // make toc panel and populate with entries (links to document
            // sections)
            const panel = makePanel();
            if (!panel)
                return;
            makeEntries(panel);
            // attach panel to document after making entries, so 'toc' heading
            // in panel isn't included in toc
            document.body.insertBefore(panel, document.body.firstChild);

            // initial panel state
            if (
                options.startOpen === 'true' ||
                (options.startOpen === 'auto' && !isSmallScreen())
            )
                openPanel();
            else
                closePanel();

            // attach click, scroll, and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('scroll', onScroll);
            window.addEventListener('hashchange', onScroll);
            window.addEventListener('keyup', onKeyUp);
            onScroll();

            // add class to push document body down out of way of toc button
            document.body.classList.add('toc_body_nudge');
        }

        // determine if screen wide enough to fit toc panel
        function isSmallScreen() {
            // in default theme:
            // 816px = 8.5in = width of "page" (<body>) element
            // 260px = min width of toc panel (*2 for both sides of <body>)
            return window.innerWidth < 816 + 260 * 2;
        }

        // when mouse is clicked anywhere in window
        function onClick() {
            if (isSmallScreen())
                closePanel();
        }

        // when window is scrolled or hash changed
        function onScroll() {
            highlightViewed();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            // close on esc
            if (event.key === 'Escape')
                closePanel();
        }

        // find entry of currently viewed document section in toc and highlight
        function highlightViewed() {
            const firstId = getFirstInView(options.typesQuery);

            // get toc entries (links), unhighlight all, then highlight viewed
            const list = document.getElementById('toc_list');
            if (!firstId || !list)
                return;
            const links = list.querySelectorAll('a');
            for (const link of links)
                link.dataset.viewing = 'false';
            const link = list.querySelector('a[href="#' + firstId + '"]');
            if (!link)
                return;
            link.dataset.viewing = 'true';
        }

        // get first or previous toc listed element in top half of view
        function getFirstInView(query) {
            // get all elements matching query and with id
            const elements = document.querySelectorAll(query);
            const elementsWithIds = [];
            for (const element of elements) {
                if (element.id)
                    elementsWithIds.push(element);
            }


            // get first or previous element in top half of view
            for (let i = 0; i < elementsWithIds.length; i++) {
                const element = elementsWithIds[i];
                const prevElement = elementsWithIds[Math.max(0, i - 1)];
                if (element.getBoundingClientRect().top >= 0) {
                    if (
                        element.getBoundingClientRect().top <
                        window.innerHeight / 2
                    )
                        return element.id;
                    else
                        return prevElement.id;
                }
            }
        }

        // make panel
        function makePanel() {
            // create panel
            const panel = document.createElement('div');
            panel.id = 'toc_panel';
            if (options.bullets === 'true')
                panel.dataset.bullets = 'true';

            // create header
            const header = document.createElement('div');
            header.id = 'toc_header';

            // create toc button
            const button = document.createElement('button');
            button.id = 'toc_button';
            button.innerHTML = document.querySelector('.icon_th_list').innerHTML;
            button.title = 'Table of Contents';
            button.classList.add('icon_button');

            // create header text
            const text = document.createElement('h4');
            text.innerHTML = 'Table of Contents';

            // create container for toc list
            const list = document.createElement('div');
            list.id = 'toc_list';

            // attach click listeners
            panel.addEventListener('click', onPanelClick);
            header.addEventListener('click', onHeaderClick);
            button.addEventListener('click', onButtonClick);

            // attach elements
            header.appendChild(button);
            header.appendChild(text);
            panel.appendChild(header);
            panel.appendChild(list);

            return panel;
        }

        // create toc entries (links) to each element of the specified types
        function makeEntries(panel) {
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements) {
                // do not add link if element doesn't have assigned id
                if (!element.id)
                    continue;

                // create link/list item
                const link = document.createElement('a');
                link.classList.add('toc_link');
                switch (element.tagName.toLowerCase()) {
                    case 'h1':
                        link.dataset.level = '1';
                        break;
                    case 'h2':
                        link.dataset.level = '2';
                        break;
                    case 'h3':
                        link.dataset.level = '3';
                        break;
                    case 'h4':
                        link.dataset.level = '4';
                        break;
                }
                link.title = element.innerText;
                let text = element.innerText;
                if (text.length > options.charLimit)
                    text = text.slice(0, options.charLimit) + '...';
                link.innerHTML = text;
                link.href = '#' + element.id;
                link.addEventListener('click', onLinkClick);

                // attach link
                panel.querySelector('#toc_list').appendChild(link);
            }
        }

        // when panel is clicked
        function onPanelClick(event) {
            // stop click from propagating to window/document and closing panel
            event.stopPropagation();
        }

        // when header itself is clicked
        function onHeaderClick(event) {
            togglePanel();
        }

        // when button is clicked
        function onButtonClick(event) {
            togglePanel();
            // stop header underneath button from also being clicked
            event.stopPropagation();
        }

        // when link is clicked
        function onLinkClick(event) {
            if (
                options.clickClose === 'true' ||
                (options.clickClose === 'auto' && isSmallScreen())
            )
                closePanel();
            else
                openPanel();
        }

        // open panel if closed, close if opened
        function togglePanel() {
            const panel = document.getElementById('toc_panel');
            if (!panel)
                return;

            if (panel.dataset.open === 'true')
                closePanel();
            else
                openPanel();
        }

        // open panel
        function openPanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'true';
        }

        // close panel
        function closePanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'false';
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- th list icon -->

<template class="icon_th_list">
    <!-- modified from: https://fontawesome.com/icons/th-list -->
    <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
        <path
            fill="currentColor"
            d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            tabindex="-1"
        ></path>
    </svg>
</template>
<!-- lightbox plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user clicks on an
        // image, the image fills the screen and the user can pan/drag/zoom
        // the image and navigate between other images in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'lightbox';

        // default plugin options
        const options = {
            // list of possible zoom/scale factors
            zoomSteps:
                '0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1,' +
                '1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8',
            // whether to fit image to view ('fit'), display at 100% and shrink
            // if necessary ('shrink'), or always display at 100% ('100')
            defaultZoom: 'fit',
            // whether to zoom in/out toward center of view ('true') or mouse
            // ('false')
            centerZoom: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <img> element
            const imgs = document.querySelectorAll('figure > img');
            let count = 1;
            for (const img of imgs) {
                img.classList.add('lightbox_document_img');
                img.dataset.number = count;
                img.dataset.total = imgs.length;
                img.addEventListener('click', openLightbox);
                count++;
            }

            // attach mouse and key listeners to window
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('keyup', onKeyUp);
        }

        // when mouse is moved anywhere in window
        function onWindowMouseMove(event) {
            window.mouseX = event.clientX;
            window.mouseY = event.clientY;
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'lightbox_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'lightbox_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeLightbox();
                    break;
            }
        }

        // open lightbox
        function openLightbox() {
            const lightbox = makeLightbox(this);
            if (!lightbox)
                return;

            blurBody(lightbox);
            document.body.appendChild(lightbox);
        }

        // make lightbox
        function makeLightbox(img) {
            // delete lightbox if it exists, start fresh
            closeLightbox();

            // create screen overlay containing lightbox
            const overlay = document.createElement('div');
            overlay.id = 'lightbox_overlay';

            // create image info boxes
            const numberInfo = document.createElement('div');
            const zoomInfo = document.createElement('div');
            numberInfo.id = 'lightbox_number_info';
            zoomInfo.id = 'lightbox_zoom_info';

            // create container for image
            const imageContainer = document.createElement('div');
            imageContainer.id = 'lightbox_image_container';
            const lightboxImg = makeLightboxImg(
                img,
                imageContainer,
                numberInfo,
                zoomInfo
            );
            imageContainer.appendChild(lightboxImg);

            // create bottom container for caption and navigation buttons
            const bottomContainer = document.createElement('div');
            bottomContainer.id = 'lightbox_bottom_container';
            const caption = makeCaption(img);
            const prevButton = makePrevButton(img);
            const nextButton = makeNextButton(img);
            bottomContainer.appendChild(prevButton);
            bottomContainer.appendChild(caption);
            bottomContainer.appendChild(nextButton);

            // attach top middle and bottom to overlay
            overlay.appendChild(numberInfo);
            overlay.appendChild(zoomInfo);
            overlay.appendChild(imageContainer);
            overlay.appendChild(bottomContainer);

            return overlay;
        }

        // make <img> object that is intuitively draggable and zoomable
        function makeLightboxImg(
            sourceImg,
            container,
            numberInfoBox,
            zoomInfoBox
        ) {
            // create copy of source <img>
            const img = sourceImg.cloneNode(true);
            img.classList.remove('lightbox_document_img');
            img.removeAttribute('id');
            img.removeAttribute('width');
            img.removeAttribute('height');
            img.style.position = 'unset';
            img.style.margin = '0';
            img.style.padding = '0';
            img.style.width = '';
            img.style.height = '';
            img.style.minWidth = '';
            img.style.minHeight = '';
            img.style.maxWidth = '';
            img.style.maxHeight = '';
            img.id = 'lightbox_img';

            // build sorted list of unique zoomSteps, always including a 100%
            let zoomSteps = [];
            const optionsZooms = options.zoomSteps.split(/[^0-9.]/);
            for (const optionZoom of optionsZooms) {
                const newZoom = parseFloat(optionZoom);
                if (newZoom && !zoomSteps.includes(newZoom))
                    zoomSteps.push(newZoom);
            }
            if (!zoomSteps.includes(1))
                zoomSteps.push(1);
            zoomSteps = zoomSteps.sort(function sortNumber(a, b) {
                return a - b;
            });

            // <img> object property variables
            let zoom = 1;
            let translateX = 0;
            let translateY = 0;
            let clickMouseX = undefined;
            let clickMouseY = undefined;
            let clickTranslateX = undefined;
            let clickTranslateY = undefined;

            updateNumberInfo();

            // update image numbers displayed in info box
            function updateNumberInfo() {
                numberInfoBox.innerHTML =
                    sourceImg.dataset.number + ' of ' + sourceImg.dataset.total;
            }

            // update zoom displayed in info box
            function updateZoomInfo() {
                let zoomInfo = zoom * 100;
                if (!Number.isInteger(zoomInfo))
                    zoomInfo = zoomInfo.toFixed(2);
                zoomInfoBox.innerHTML = zoomInfo + '%';
            }

            // move to closest zoom step above current zoom
            const zoomIn = function() {
                for (const zoomStep of zoomSteps) {
                    if (zoomStep > zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                updateTransform();
            };

            // move to closest zoom step above current zoom
            const zoomOut = function() {
                zoomSteps.reverse();
                for (const zoomStep of zoomSteps) {
                    if (zoomStep < zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                zoomSteps.reverse();

                updateTransform();
            };

            // update display of <img> based on scale/translate properties
            const updateTransform = function() {
                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                // get new width/height after scale
                const rect = img.getBoundingClientRect();
                // limit translate
                translateX = Math.max(translateX, -rect.width / 2);
                translateX = Math.min(translateX, rect.width / 2);
                translateY = Math.max(translateY, -rect.height / 2);
                translateY = Math.min(translateY, rect.height / 2);

                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                updateZoomInfo();
            };

            // fit <img> to container
            const fit = function() {
                // no x/y offset, 100% zoom by default
                translateX = 0;
                translateY = 0;
                zoom = 1;

                // widths of <img> and container
                const imgWidth = img.naturalWidth;
                const imgHeight = img.naturalHeight;
                const containerWidth = parseFloat(
                    window.getComputedStyle(container).width
                );
                const containerHeight = parseFloat(
                    window.getComputedStyle(container).height
                );

                // how much zooming is needed to fit <img> to container
                const xRatio = imgWidth / containerWidth;
                const yRatio = imgHeight / containerHeight;
                const maxRatio = Math.max(xRatio, yRatio);
                const newZoom = 1 / maxRatio;

                // fit <img> to container according to option
                if (options.defaultZoom === 'shrink') {
                    if (maxRatio > 1)
                        zoom = newZoom;
                } else if (options.defaultZoom === 'fit')
                    zoom = newZoom;

                updateTransform();
            };

            // when mouse wheel is rolled anywhere in container
            const onContainerWheel = function(event) {
                if (!event)
                    return;

                // let ctrl + mouse wheel to zoom behave as normal
                if (event.ctrlKey)
                    return;

                // prevent normal scroll behavior
                event.preventDefault();
                event.stopPropagation();

                // point around which to scale img
                const viewRect = container.getBoundingClientRect();
                const viewX = (viewRect.left + viewRect.right) / 2;
                const viewY = (viewRect.top + viewRect.bottom) / 2;
                const originX = options.centerZoom === 'true' ? viewX : mouseX;
                const originY = options.centerZoom === 'true' ? viewY : mouseY;

                // get point on image under origin
                const oldRect = img.getBoundingClientRect();
                const oldPercentX = (originX - oldRect.left) / oldRect.width;
                const oldPercentY = (originY - oldRect.top) / oldRect.height;

                // increment/decrement zoom
                if (event.deltaY < 0)
                    zoomIn();
                if (event.deltaY > 0)
                    zoomOut();

                // get offset between previous image point and origin
                const newRect = img.getBoundingClientRect();
                const offsetX =
                    originX - (newRect.left + newRect.width * oldPercentX);
                const offsetY =
                    originY - (newRect.top + newRect.height * oldPercentY);

                // translate image to keep image point under origin
                translateX += offsetX;
                translateY += offsetY;

                // perform translate
                updateTransform();
            };

            // when container is clicked
            function onContainerClick(event) {
                // if container itself is target of click, and not other
                // element above it
                if (event.target === this)
                    closeLightbox();
            }

            // when mouse button is pressed on image
            const onImageMouseDown = function(event) {
                // store original mouse position relative to image
                clickMouseX = window.mouseX;
                clickMouseY = window.mouseY;
                clickTranslateX = translateX;
                clickTranslateY = translateY;
                event.stopPropagation();
                event.preventDefault();
            };

            // when mouse button is released anywhere in window
            const onWindowMouseUp = function(event) {
                // reset original mouse position
                clickMouseX = undefined;
                clickMouseY = undefined;
                clickTranslateX = undefined;
                clickTranslateY = undefined;

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mouseup', onWindowMouseUp);
            };

            // when mouse is moved anywhere in window
            const onWindowMouseMove = function(event) {
                if (
                    clickMouseX === undefined ||
                    clickMouseY === undefined ||
                    clickTranslateX === undefined ||
                    clickTranslateY === undefined
                )
                    return;

                // offset image based on original and current mouse position
                translateX = clickTranslateX + window.mouseX - clickMouseX;
                translateY = clickTranslateY + window.mouseY - clickMouseY;
                updateTransform();
                event.preventDefault();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mousemove', onWindowMouseMove);
            };

            // when window is resized
            const onWindowResize = function(event) {
                fit();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('resize', onWindowResize);
            };

            // attach the necessary event listeners
            img.addEventListener('dblclick', fit);
            img.addEventListener('mousedown', onImageMouseDown);
            container.addEventListener('wheel', onContainerWheel);
            container.addEventListener('mousedown', onContainerClick);
            container.addEventListener('touchstart', onContainerClick);
            window.addEventListener('mouseup', onWindowMouseUp);
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('resize', onWindowResize);

            // run fit() after lightbox atttached to document and <img> Loaded
            // so needed container and img dimensions available
            img.addEventListener('load', fit);

            return img;
        }

        // make caption
        function makeCaption(img) {
            const caption = document.createElement('div');
            caption.id = 'lightbox_caption';
            const captionSource = img.nextElementSibling;
            if (captionSource.tagName.toLowerCase() === 'figcaption') {
                const captionCopy = makeCopy(captionSource);
                caption.innerHTML = captionCopy.innerHTML;
            }

            caption.addEventListener('touchstart', function(event) {
                event.stopPropagation();
            });

            return caption;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // make button to jump to previous image in document
        function makePrevButton(img) {
            const prevButton = document.createElement('button');
            prevButton.id = 'lightbox_prev_button';
            prevButton.title = 'Jump to the previous image in the document [←]';
            prevButton.classList.add('icon_button', 'lightbox_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;

            // attach click listeners to button
            prevButton.addEventListener('click', function() {
                getPrevImg(img).click();
            });

            return prevButton;
        }

        // make button to jump to next image in document
        function makeNextButton(img) {
            const nextButton = document.createElement('button');
            nextButton.id = 'lightbox_next_button';
            nextButton.title = 'Jump to the next image in the document [→]';
            nextButton.classList.add('icon_button', 'lightbox_button');
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;

            // attach click listeners to button
            nextButton.addEventListener('click', function() {
                getNextImg(img).click();
            });

            return nextButton;
        }

        // get previous image in document
        function getPrevImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if < 1
            if (index - 1 >= 0)
                index--;
            else
                index = imgs.length - 1;
            return imgs[index];
        }

        // get next image in document
        function getNextImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if > total
            if (index + 1 <= imgs.length - 1)
                index++;
            else
                index = 0;
            return imgs[index];
        }

        // close lightbox
        function closeLightbox() {
            focusBody();

            const lightbox = document.getElementById('lightbox_overlay');
            if (lightbox)
                lightbox.remove();
        }

        // make all elements behind lightbox non-focusable
        function blurBody(overlay) {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.tabIndex = -1;
            document.body.classList.add('body_no_scroll');
        }

        // make all elements focusable again
        function focusBody() {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.removeAttribute('tabIndex');
            document.body.classList.remove('body_no_scroll');
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- attributes plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows arbitrary HTML attributes to be attached
        // to (almost) any element. Place an HTML comment inside or next to the
        // desired element in the format <!-- $attribute="value" -->

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'attributes';

        // default plugin options
        const options = {
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // get list of comments in document
            const comments = findComments();

            for(const comment of comments)
                if (comment.parentElement)
                    addAttributes(
                        comment.parentElement,
                        comment.nodeValue.trim()
                    );
        }

        // add html attributes to specified element based on string of 
        // html attributes and values
        function addAttributes(element, text) {
            // regex's for finding attribute/value pairs in the format of
            // attribute="value" or attribute='value
            const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
            const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

            // loop through attribute/value pairs
            let match;
            while(match = text.match(regex2) || text.match(regex1)) {
                // get attribute and value from regex capture groups
                let attribute = match[1];
                let value = match[2];

                // remove from string
                text = text.substring(match.index + match[0].length);

                if (!attribute || !value)
                    break;

                // set attribute of parent element
                try {
                    element.setAttribute(attribute, value);
                } catch(error) {
                    console.log(error);
                }

                // special case for colspan
                if (attribute === 'colspan')
                    removeTableCells(element, value);
            }
        }

        // get list of comment elements in document
        function findComments() {
            const comments = [];

            // iterate over comment nodes in document
            function acceptNode(node) {
                return NodeFilter.FILTER_ACCEPT;
            }
            const iterator = document.createNodeIterator(
                document.body,
                NodeFilter.SHOW_COMMENT,
                acceptNode
            );
            let node;
            while(node = iterator.nextNode())
                comments.push(node);

            return comments;
        }

        // remove certain number of cells after specified cell
        function removeTableCells(cell, number) {
            number = parseInt(number);
            if (!number)
                return;

            // remove elements
            for(; number > 1; number--) {
                if (cell.nextElementSibling)
                    cell.nextElementSibling.remove();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<!-- mathjax plugin configuration -->

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        "CommonHTML": { linebreaks: { automatic: true } },
        "HTML-CSS": { linebreaks: { automatic: true } },
        "SVG": { linebreaks: { automatic: true } },
        "fast-preview": { disabled: true }
  });
</script>

<!-- mathjax plugin -->

<script
    src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
    integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
    crossorigin="anonymous"
>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'MathJax' allows the proper rendering of
    // math/equations written in LaTeX.

    // https://www.mathjax.org/
</script>
<!-- annotations plugin -->

<script>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'Hypothesis' allows public annotation of the
    // manuscript.

    // https://web.hypothes.is/

    // plugin configuration
    window.hypothesisConfig = function() {
        return {
            branding: {
                accentColor: '#2196f3',
                appBackgroundColor: '#f8f8f8',
                ctaBackgroundColor: '#f8f8f8',
                ctaTextColor: '#000000',
                selectionFontFamily: 'Open Sans, Helvetica, sans serif',
                annotationFontFamily: 'Open Sans, Helvetica, sans serif'
            }
        };
    };

    // hypothesis client script
    const embed = 'https://hypothes.is/embed.js';
    // hypothesis annotation count query url
    const query = 'https://api.hypothes.is/api/search?limit=0&url='

    
    // start script
    function start() {
        const button = makeButton();
        document.body.insertBefore(button, document.body.firstChild);
        insertCount(button);
    }

    // make button
    function makeButton() {
        // create button
        const button = document.createElement('button');
        button.id = 'hypothesis_button';
        button.innerHTML = document.querySelector('.icon_hypothesis').innerHTML;
        button.title = 'Hypothesis annotations';
        button.classList.add('icon_button');

        function onClick(event) {
            onButtonClick(event, button);
        }

        // attach click listeners
        button.addEventListener('click', onClick);

        return button;
    }

    // insert annotations count
    async function insertCount(button) {
        // get annotation count from Hypothesis based on url
        let count = '-';
        try {
            const canonical = document.querySelector('link[rel="canonical"]');
            const location = window.location;
            const url = encodeURIComponent((canonical || location).href);
            const response = await fetch(query + url);
            const json = await response.json();
            count = json.total || '-';
        } catch(error) {
            console.log(error);
        }
        
        // put count into button
        const counter = document.createElement('span');
        counter.id = 'hypothesis_count';
        counter.innerHTML = count;
        button.title = 'View ' + count + ' Hypothesis annotations';
        button.append(counter);
    }

    // when button is clicked
    function onButtonClick(event, button) {
        const script = document.createElement('script');
        script.src = embed;
        document.body.append(script);
        button.remove();
    }

    window.addEventListener('load', start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
    <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
    <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
        <path
            fill="currentColor"
            d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
            tabindex="-1"
        ></path>
    </svg>
</template>
<!-- analytics plugin -->

<!-- copy and paste code from Google Analytics or similar service here -->
</body>
</html>
